
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Field guide for biomarker discovery from omics data in aquaculture">
      
      
        <meta name="author" content="Shelly A. Wanamaker">
      
      
        <link rel="canonical" href="https://resilience-biomarkers-for-aquaculture.github.io/field-guide/pipelines/validation/">
      
      
        <link rel="prev" href="../classifier-path/">
      
      
        <link rel="next" href="../../timeline/">
      
      
        
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.7.1">
    
    
      
        <title>Validation & Pitfalls - Resilience Biomarkers Field Guide</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.484c7ddc.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.ab4e12ef.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="blue" data-md-color-accent="light-blue">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#validation-pitfalls" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Resilience Biomarkers Field Guide" class="md-header__button md-logo" aria-label="Resilience Biomarkers Field Guide" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Resilience Biomarkers Field Guide
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Validation & Pitfalls
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="blue" data-md-color-accent="light-blue"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="blue" data-md-color-accent="light-blue"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../.." class="md-tabs__link">
        
  
  
    
  
  Start Here

      </a>
    </li>
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../../problem-framing/" class="md-tabs__link">
        
  
  
    
  
  Problem Framing

      </a>
    </li>
  

      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../decision-tree/" class="md-tabs__link">
          
  
  
  Pipelines

        </a>
      </li>
    
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../../timeline/" class="md-tabs__link">
        
  
  
    
  
  Year in Review

      </a>
    </li>
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../sources/issues-index/" class="md-tabs__link">
          
  
  
  Sources

        </a>
      </li>
    
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../../glossary/" class="md-tabs__link">
        
  
  
    
  
  Glossary

      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Resilience Biomarkers Field Guide" class="md-nav__button md-logo" aria-label="Resilience Biomarkers Field Guide" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    Resilience Biomarkers Field Guide
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Start Here
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../problem-framing/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Problem Framing
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
      
        
        
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" checked>
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    Pipelines
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Pipelines
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../decision-tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Decision Tree
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../classifier-path/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Two-Step Classifier
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    Validation & Pitfalls
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    Validation & Pitfalls
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" class="md-nav__link">
    <span class="md-ellipsis">
      
        Overview
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#validation-strategies" class="md-nav__link">
    <span class="md-ellipsis">
      
        Validation Strategies
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Validation Strategies">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-leave-one-study-out-loso-cross-validation" class="md-nav__link">
    <span class="md-ellipsis">
      
        1. Leave-One-Study-Out (LOSO) Cross-Validation
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-within-study-cross-validation" class="md-nav__link">
    <span class="md-ellipsis">
      
        2. Within-Study Cross-Validation
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-independent-validation-cohort" class="md-nav__link">
    <span class="md-ellipsis">
      
        3. Independent Validation Cohort
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#common-pitfalls" class="md-nav__link">
    <span class="md-ellipsis">
      
        Common Pitfalls
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Common Pitfalls">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-data-leakage" class="md-nav__link">
    <span class="md-ellipsis">
      
        1. Data Leakage
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="1. Data Leakage">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#leakage-scenario-1-normalization-across-trainingtest" class="md-nav__link">
    <span class="md-ellipsis">
      
        Leakage Scenario 1: Normalization Across Training+Test
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#leakage-scenario-2-feature-selection-on-all-data" class="md-nav__link">
    <span class="md-ellipsis">
      
        Leakage Scenario 2: Feature Selection on All Data
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-batch-effects" class="md-nav__link">
    <span class="md-ellipsis">
      
        2. Batch Effects
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="2. Batch Effects">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#batch-correction-attempts" class="md-nav__link">
    <span class="md-ellipsis">
      
        Batch Correction Attempts
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#post-data-integration-solution" class="md-nav__link">
    <span class="md-ellipsis">
      
        Post-Data Integration Solution
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-normalization-timing-method" class="md-nav__link">
    <span class="md-ellipsis">
      
        3. Normalization Timing &amp; Method
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="3. Normalization Timing &amp; Method">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#understanding-the-pipeline" class="md-nav__link">
    <span class="md-ellipsis">
      
        Understanding the Pipeline
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#normalization-methods" class="md-nav__link">
    <span class="md-ellipsis">
      
        Normalization Methods
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4-oversimplified-phenotype-definitions" class="md-nav__link">
    <span class="md-ellipsis">
      
        4. Oversimplified Phenotype Definitions
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#5-small-sample-sizes" class="md-nav__link">
    <span class="md-ellipsis">
      
        5. Small Sample Sizes
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#6-innate-vs-reactive-biomarkers" class="md-nav__link">
    <span class="md-ellipsis">
      
        6. Innate vs. Reactive Biomarkers
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#validation-checklist" class="md-nav__link">
    <span class="md-ellipsis">
      
        Validation Checklist
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#troubleshooting-guide" class="md-nav__link">
    <span class="md-ellipsis">
      
        Troubleshooting Guide
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Troubleshooting Guide">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#poor-loso-performance" class="md-nav__link">
    <span class="md-ellipsis">
      
        Poor LOSO Performance
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genes-dont-validate" class="md-nav__link">
    <span class="md-ellipsis">
      
        Genes Don't Validate
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#suspect-data-leakage" class="md-nav__link">
    <span class="md-ellipsis">
      
        Suspect Data Leakage
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#related-resources" class="md-nav__link">
    <span class="md-ellipsis">
      
        Related Resources
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../timeline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Year in Review
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Sources
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            
  
    Sources
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../sources/issues-index/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    GitHub Issues
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../sources/posts-index/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Notebook Posts
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../glossary/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Glossary
  

    
  </span>
  
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" class="md-nav__link">
    <span class="md-ellipsis">
      
        Overview
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#validation-strategies" class="md-nav__link">
    <span class="md-ellipsis">
      
        Validation Strategies
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Validation Strategies">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-leave-one-study-out-loso-cross-validation" class="md-nav__link">
    <span class="md-ellipsis">
      
        1. Leave-One-Study-Out (LOSO) Cross-Validation
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-within-study-cross-validation" class="md-nav__link">
    <span class="md-ellipsis">
      
        2. Within-Study Cross-Validation
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-independent-validation-cohort" class="md-nav__link">
    <span class="md-ellipsis">
      
        3. Independent Validation Cohort
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#common-pitfalls" class="md-nav__link">
    <span class="md-ellipsis">
      
        Common Pitfalls
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Common Pitfalls">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-data-leakage" class="md-nav__link">
    <span class="md-ellipsis">
      
        1. Data Leakage
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="1. Data Leakage">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#leakage-scenario-1-normalization-across-trainingtest" class="md-nav__link">
    <span class="md-ellipsis">
      
        Leakage Scenario 1: Normalization Across Training+Test
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#leakage-scenario-2-feature-selection-on-all-data" class="md-nav__link">
    <span class="md-ellipsis">
      
        Leakage Scenario 2: Feature Selection on All Data
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-batch-effects" class="md-nav__link">
    <span class="md-ellipsis">
      
        2. Batch Effects
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="2. Batch Effects">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#batch-correction-attempts" class="md-nav__link">
    <span class="md-ellipsis">
      
        Batch Correction Attempts
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#post-data-integration-solution" class="md-nav__link">
    <span class="md-ellipsis">
      
        Post-Data Integration Solution
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-normalization-timing-method" class="md-nav__link">
    <span class="md-ellipsis">
      
        3. Normalization Timing &amp; Method
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="3. Normalization Timing &amp; Method">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#understanding-the-pipeline" class="md-nav__link">
    <span class="md-ellipsis">
      
        Understanding the Pipeline
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#normalization-methods" class="md-nav__link">
    <span class="md-ellipsis">
      
        Normalization Methods
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4-oversimplified-phenotype-definitions" class="md-nav__link">
    <span class="md-ellipsis">
      
        4. Oversimplified Phenotype Definitions
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#5-small-sample-sizes" class="md-nav__link">
    <span class="md-ellipsis">
      
        5. Small Sample Sizes
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#6-innate-vs-reactive-biomarkers" class="md-nav__link">
    <span class="md-ellipsis">
      
        6. Innate vs. Reactive Biomarkers
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#validation-checklist" class="md-nav__link">
    <span class="md-ellipsis">
      
        Validation Checklist
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#troubleshooting-guide" class="md-nav__link">
    <span class="md-ellipsis">
      
        Troubleshooting Guide
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Troubleshooting Guide">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#poor-loso-performance" class="md-nav__link">
    <span class="md-ellipsis">
      
        Poor LOSO Performance
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genes-dont-validate" class="md-nav__link">
    <span class="md-ellipsis">
      
        Genes Don't Validate
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#suspect-data-leakage" class="md-nav__link">
    <span class="md-ellipsis">
      
        Suspect Data Leakage
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#related-resources" class="md-nav__link">
    <span class="md-ellipsis">
      
        Related Resources
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              
              <article class="md-content__inner md-typeset">
                
                  



<h1 id="validation-pitfalls">Validation &amp; Pitfalls</h1>
<h2 id="overview">Overview</h2>
<p>Validation is critical for ensuring biomarker panels generalize beyond the discovery cohort. This guide covers validation strategies and common pitfalls encountered in this project, with emphasis on what actually went wrong and how to avoid it.</p>
<hr />
<h2 id="validation-strategies">Validation Strategies</h2>
<h3 id="1-leave-one-study-out-loso-cross-validation">1. Leave-One-Study-Out (LOSO) Cross-Validation</h3>
<p><strong>What it is:</strong> Train classifier on all datasets except one, test on the held-out dataset</p>
<p><strong>Why it's essential:</strong></p>
<ul>
<li>Tests <strong>cross-study generalization</strong></li>
<li>Avoids overfitting to study-specific effects</li>
<li>Simulates real-world scenario: predict phenotypes in new, unseen studies</li>
</ul>
<p><strong>Implementation:</strong></p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="n">datasets</span> <span class="o">=</span> <span class="p">[</span><span class="n">dataset1</span><span class="p">,</span> <span class="n">dataset2</span><span class="p">,</span> <span class="n">dataset3</span><span class="p">,</span> <span class="n">dataset4</span><span class="p">,</span> <span class="n">dataset5</span><span class="p">]</span>
<a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a>
<a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a><span class="k">for</span> <span class="n">test_idx</span><span class="p">,</span> <span class="n">test_dataset</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">datasets</span><span class="p">):</span>
<a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a>    <span class="c1"># Training set: all except test_dataset</span>
<a id="__codelineno-0-5" name="__codelineno-0-5" href="#__codelineno-0-5"></a>    <span class="n">train_datasets</span> <span class="o">=</span> <span class="p">[</span><span class="n">ds</span> <span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">ds</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">datasets</span><span class="p">)</span> <span class="k">if</span> <span class="n">i</span> <span class="o">!=</span> <span class="n">test_idx</span><span class="p">]</span>
<a id="__codelineno-0-6" name="__codelineno-0-6" href="#__codelineno-0-6"></a>
<a id="__codelineno-0-7" name="__codelineno-0-7" href="#__codelineno-0-7"></a>    <span class="n">X_train</span> <span class="o">=</span> <span class="n">combine_expression_data</span><span class="p">(</span><span class="n">train_datasets</span><span class="p">,</span> <span class="n">gene_panel</span><span class="p">)</span>
<a id="__codelineno-0-8" name="__codelineno-0-8" href="#__codelineno-0-8"></a>    <span class="n">y_train</span> <span class="o">=</span> <span class="n">combine_labels</span><span class="p">(</span><span class="n">train_datasets</span><span class="p">)</span>
<a id="__codelineno-0-9" name="__codelineno-0-9" href="#__codelineno-0-9"></a>
<a id="__codelineno-0-10" name="__codelineno-0-10" href="#__codelineno-0-10"></a>    <span class="c1"># Fit classifier</span>
<a id="__codelineno-0-11" name="__codelineno-0-11" href="#__codelineno-0-11"></a>    <span class="n">classifier</span><span class="o">.</span><span class="n">fit</span><span class="p">(</span><span class="n">X_train</span><span class="p">,</span> <span class="n">y_train</span><span class="p">)</span>
<a id="__codelineno-0-12" name="__codelineno-0-12" href="#__codelineno-0-12"></a>
<a id="__codelineno-0-13" name="__codelineno-0-13" href="#__codelineno-0-13"></a>    <span class="c1"># Test on held-out study</span>
<a id="__codelineno-0-14" name="__codelineno-0-14" href="#__codelineno-0-14"></a>    <span class="n">X_test</span> <span class="o">=</span> <span class="n">test_dataset</span><span class="o">.</span><span class="n">expression</span><span class="p">[</span><span class="n">gene_panel</span><span class="p">]</span>
<a id="__codelineno-0-15" name="__codelineno-0-15" href="#__codelineno-0-15"></a>    <span class="n">y_test</span> <span class="o">=</span> <span class="n">test_dataset</span><span class="o">.</span><span class="n">labels</span>
<a id="__codelineno-0-16" name="__codelineno-0-16" href="#__codelineno-0-16"></a>
<a id="__codelineno-0-17" name="__codelineno-0-17" href="#__codelineno-0-17"></a>    <span class="n">accuracy</span> <span class="o">=</span> <span class="n">classifier</span><span class="o">.</span><span class="n">score</span><span class="p">(</span><span class="n">X_test</span><span class="p">,</span> <span class="n">y_test</span><span class="p">)</span>
<a id="__codelineno-0-18" name="__codelineno-0-18" href="#__codelineno-0-18"></a>    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;LOSO Fold </span><span class="si">{</span><span class="n">test_idx</span><span class="o">+</span><span class="mi">1</span><span class="si">}</span><span class="s2"> (test=</span><span class="si">{</span><span class="n">test_dataset</span><span class="o">.</span><span class="n">name</span><span class="si">}</span><span class="s2">): </span><span class="si">{</span><span class="n">accuracy</span><span class="si">:</span><span class="s2">.3f</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
</code></pre></div>
<p><strong>Interpretation:</strong></p>
<ul>
<li>✅ <strong>Good:</strong> Consistent accuracy (e.g., 0.80-0.90) across all folds</li>
<li>⚠️ <strong>Warning:</strong> One fold significantly lower → investigate study-specific effects</li>
<li>❌ <strong>Poor:</strong> High variance in accuracy or &lt; 0.70 average → panel doesn't generalize</li>
</ul>
<div class="admonition warning">
<p class="admonition-title">Big Lesson #4: Training Set in Test Set</p>
<p><strong>Never include the training study in the test set</strong> when evaluating cross-study performance. This leads to data leakage and falsely optimistic results.</p>
<p>✅ Correct: LOSO (train on studies 1,2,3; test on study 4)<br />
❌ Wrong: Train on studies 1,2,3,4; test on studies 1,2,3,4</p>
</div>
<p><strong>Related:</strong> <a href="https://github.com/Resilience-Biomarkers-for-Aquaculture/Cvirg_Pmarinus_RNAseq/issues/44">Issue #44</a></p>
<h3 id="2-within-study-cross-validation">2. Within-Study Cross-Validation</h3>
<p><strong>What it is:</strong> Hold out samples within a single study (k-fold or train/test split)</p>
<p><strong>When to use:</strong></p>
<ul>
<li>Initial model development and parameter tuning</li>
<li>Assessing overfitting within a study</li>
<li>When you only have one study available (but be cautious about generalizability)</li>
</ul>
<p><strong>Limitations:</strong></p>
<ul>
<li>Does NOT test cross-study generalization</li>
<li>May overestimate performance if study-specific effects are strong</li>
<li>Use as a first step, but follow with LOSO</li>
</ul>
<h3 id="3-independent-validation-cohort">3. Independent Validation Cohort</h3>
<p><strong>Gold standard:</strong> Test on completely independent dataset never used during development</p>
<p><strong>Challenges:</strong></p>
<ul>
<li>Requires finding additional suitable datasets</li>
<li>May have different experimental designs or phenotype definitions</li>
<li>Labor-intensive to process and integrate</li>
</ul>
<p><strong>Status in this project:</strong> <a href="https://github.com/Resilience-Biomarkers-for-Aquaculture/Cvirg_Pmarinus_RNAseq/issues/54">Issue #54</a> (find more datasets) - postponed</p>
<hr />
<h2 id="common-pitfalls">Common Pitfalls</h2>
<h3 id="1-data-leakage">1. Data Leakage</h3>
<p><strong>Definition:</strong> Information from test set "leaks" into training process, causing overly optimistic performance estimates</p>
<h4 id="leakage-scenario-1-normalization-across-trainingtest">Leakage Scenario 1: Normalization Across Training+Test</h4>
<p>❌ <strong>Wrong:</strong></p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a><span class="c1"># Normalize all data together</span>
<a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a><span class="n">all_data_normalized</span> <span class="o">=</span> <span class="n">vst_transform</span><span class="p">(</span><span class="n">combine</span><span class="p">(</span><span class="n">train_data</span><span class="p">,</span> <span class="n">test_data</span><span class="p">))</span>
<a id="__codelineno-1-3" name="__codelineno-1-3" href="#__codelineno-1-3"></a>
<a id="__codelineno-1-4" name="__codelineno-1-4" href="#__codelineno-1-4"></a><span class="c1"># Then split</span>
<a id="__codelineno-1-5" name="__codelineno-1-5" href="#__codelineno-1-5"></a><span class="n">X_train</span> <span class="o">=</span> <span class="n">all_data_normalized</span><span class="p">[</span><span class="n">train_indices</span><span class="p">]</span>
<a id="__codelineno-1-6" name="__codelineno-1-6" href="#__codelineno-1-6"></a><span class="n">X_test</span> <span class="o">=</span> <span class="n">all_data_normalized</span><span class="p">[</span><span class="n">test_indices</span><span class="p">]</span>  <span class="c1"># LEAKAGE!</span>
</code></pre></div>
<p>✅ <strong>Correct:</strong></p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a><span class="c1"># Fit normalization on training data only</span>
<a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a><span class="n">vst_params</span> <span class="o">=</span> <span class="n">fit_vst</span><span class="p">(</span><span class="n">train_data</span><span class="p">)</span>
<a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a>
<a id="__codelineno-2-4" name="__codelineno-2-4" href="#__codelineno-2-4"></a><span class="c1"># Apply to train and test separately</span>
<a id="__codelineno-2-5" name="__codelineno-2-5" href="#__codelineno-2-5"></a><span class="n">X_train</span> <span class="o">=</span> <span class="n">apply_vst</span><span class="p">(</span><span class="n">train_data</span><span class="p">,</span> <span class="n">vst_params</span><span class="p">)</span>
<a id="__codelineno-2-6" name="__codelineno-2-6" href="#__codelineno-2-6"></a><span class="n">X_test</span> <span class="o">=</span> <span class="n">apply_vst</span><span class="p">(</span><span class="n">test_data</span><span class="p">,</span> <span class="n">vst_params</span><span class="p">)</span>
</code></pre></div>
<h4 id="leakage-scenario-2-feature-selection-on-all-data">Leakage Scenario 2: Feature Selection on All Data</h4>
<p>❌ <strong>Wrong:</strong></p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a><span class="c1"># Select features using all data</span>
<a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a><span class="n">selected_genes</span> <span class="o">=</span> <span class="n">select_top_degs</span><span class="p">(</span><span class="n">all_data</span><span class="p">,</span> <span class="n">phenotypes</span><span class="p">)</span>  <span class="c1"># LEAKAGE!</span>
<a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a>
<a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a><span class="c1"># Then split for training/testing</span>
<a id="__codelineno-3-5" name="__codelineno-3-5" href="#__codelineno-3-5"></a><span class="n">X_train</span> <span class="o">=</span> <span class="n">train_data</span><span class="p">[</span><span class="n">selected_genes</span><span class="p">]</span>
<a id="__codelineno-3-6" name="__codelineno-3-6" href="#__codelineno-3-6"></a><span class="n">X_test</span> <span class="o">=</span> <span class="n">test_data</span><span class="p">[</span><span class="n">selected_genes</span><span class="p">]</span>
</code></pre></div>
<p>✅ <strong>Correct:</strong></p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="c1"># Select features using training data only</span>
<a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a><span class="n">selected_genes</span> <span class="o">=</span> <span class="n">select_top_degs</span><span class="p">(</span><span class="n">train_data</span><span class="p">,</span> <span class="n">train_phenotypes</span><span class="p">)</span>
<a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a>
<a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a><span class="c1"># Apply to test</span>
<a id="__codelineno-4-5" name="__codelineno-4-5" href="#__codelineno-4-5"></a><span class="n">X_train</span> <span class="o">=</span> <span class="n">train_data</span><span class="p">[</span><span class="n">selected_genes</span><span class="p">]</span>
<a id="__codelineno-4-6" name="__codelineno-4-6" href="#__codelineno-4-6"></a><span class="n">X_test</span> <span class="o">=</span> <span class="n">test_data</span><span class="p">[</span><span class="n">selected_genes</span><span class="p">]</span>
</code></pre></div>
<p><strong>In this project:</strong> LOSO properly avoids leakage by keeping test studies completely separate during training</p>
<h3 id="2-batch-effects">2. Batch Effects</h3>
<p><strong>Definition:</strong> Non-biological variation between studies due to technical differences (sequencing platform, library prep, time, lab)</p>
<div class="admonition failure">
<p class="admonition-title">Big Lessons #1 &amp; #2: When Batch Effects Dominate</p>
<ul>
<li>Study-specific effects were consistently <strong>stronger than trait effects</strong></li>
<li>Attempted corrections (COMBAT, RemoveBatchEffect) had <strong>little effect</strong> on trait separation</li>
<li><strong>Pivot:</strong> Abandoned integrated analysis in favor of post-data integration</li>
</ul>
</div>
<p><strong>Observations:</strong></p>
<ul>
<li>PCA plots showed clustering by <strong>study</strong>, not by phenotype</li>
<li>Trait-based separation was weak even after batch correction</li>
<li>Integration worked only when combining very similar studies (e.g., study 4 injected + study 5)</li>
</ul>
<h4 id="batch-correction-attempts">Batch Correction Attempts</h4>
<p><strong>Tried:</strong></p>
<ul>
<li><a href="https://github.com/Resilience-Biomarkers-for-Aquaculture/Cvirg_Pmarinus_RNAseq/issues/18">Issue #18</a>: COMBAT and RemoveBatchEffect</li>
</ul>
<p><strong>Outcome:</strong> ❌ Insufficient improvement</p>
<p><strong>Lesson:</strong> Batch correction is not a panacea. When study effects are too strong, no correction method will recover trait signal.</p>
<h4 id="post-data-integration-solution">Post-Data Integration Solution</h4>
<p>Instead of correcting batches and pooling:</p>
<ol>
<li>Analyze each study independently</li>
<li>Identify genes significant in multiple studies</li>
<li>Build classifier on reproducible genes</li>
</ol>
<p><strong>Advantage:</strong> Reproducibility across studies implicitly handles batch effects</p>
<p><strong>Related:</strong> <a href="../../timeline/#february-2025-confronting-batch-effects">Timeline - February 2025</a></p>
<h3 id="3-normalization-timing-method">3. Normalization Timing &amp; Method</h3>
<h4 id="understanding-the-pipeline">Understanding the Pipeline</h4>
<p>In <code>nf-core/differentialabundance</code>:</p>
<ol>
<li><strong>Raw counts</strong> are input</li>
<li><strong>PCAs are generated</strong> before differential abundance analysis (on raw or lightly filtered counts)</li>
<li><strong>Normalization</strong> (VST, rlog, TPM) happens during DESeq2 analysis</li>
<li><strong>Batch correction</strong> (if enabled) is applied post-normalization</li>
</ol>
<div class="admonition info">
<p class="admonition-title">Normalization Insight from July 2025</p>
<p>Understanding <strong>when</strong> normalization happens is critical for interpreting QC plots:</p>
<ul>
<li>Initial PCAs show raw/lightly-processed data</li>
<li>Don't expect trait separation in initial PCAs if normalization hasn't been applied</li>
<li>Post-analysis PCAs on normalized data are more informative</li>
</ul>
</div>
<p><strong>Related:</strong> <a href="https://github.com/Resilience-Biomarkers-for-Aquaculture/Cvirg_Pmarinus_RNAseq/issues/34">Issue #34</a>, <a href="../../timeline/#july-2025-combining-studies--understanding-normalization">Timeline - July 2025</a></p>
<h4 id="normalization-methods">Normalization Methods</h4>
<p><strong>VST (Variance-Stabilizing Transformation):</strong></p>
<ul>
<li>Recommended for visualization and clustering</li>
<li>Requires sufficient genes (&gt; 1000) for stable estimation</li>
<li>Breaks down with small gene sets (&lt; 100)</li>
</ul>
<div class="admonition danger">
<p class="admonition-title">VST with Small Gene Sets</p>
<p>In the stepwise approach, Step 1 filtering left only ~50 genes. DESeq2's VST could not work properly with such a small set.</p>
<p><strong>Symptom:</strong> Unreliable dispersion estimates, poor model fit<br />
<strong>Solution:</strong> Use alternative normalization (TMM, TPM) or abandon stepwise approach</p>
</div>
<p><strong>Related:</strong> <a href="../decision-tree/#1-deseq2-breaks-down-with-small-gene-sets">Stepwise Pipeline</a></p>
<h3 id="4-oversimplified-phenotype-definitions">4. Oversimplified Phenotype Definitions</h3>
<div class="admonition failure">
<p class="admonition-title">Big Lesson #2: Traits Were Oversimplified</p>
<p>Initial approach used generalized trait labels across studies:</p>
<ul>
<li>"Stress" vs. "Control"</li>
<li>"Resistant" vs. "Sensitive"</li>
</ul>
<p><strong>Problem:</strong> Different studies measured different stressors with different designs:</p>
<ul>
<li>Study 1: Disease challenge, time point A</li>
<li>Study 2: Temperature stress, time point B</li>
<li>Study 3: Disease challenge, time point C, different infection route</li>
</ul>
<p><strong>Result:</strong> Trait effects were <strong>much weaker</strong> than study-specific effects</p>
</div>
<p><strong>Solution:</strong> Use more specific, harmonized phenotype definitions</p>
<ul>
<li>Compare only studies with similar experimental designs</li>
<li>Within-study trait comparisons first, then assess reproducibility across studies</li>
<li>Document and respect phenotype heterogeneity</li>
</ul>
<p><strong>Related:</strong> <a href="../../problem-framing/#defining-the-phenotype">Problem Framing - Defining the Phenotype</a></p>
<h3 id="5-small-sample-sizes">5. Small Sample Sizes</h3>
<p><strong>Challenges:</strong></p>
<ul>
<li>Insufficient power to detect true DEGs</li>
<li>High variance in estimates</li>
<li>Overfitting risk (model learns noise, not signal)</li>
</ul>
<p><strong>Mitigations:</strong></p>
<ul>
<li>Use regularized models (LASSO, elastic net) to reduce overfitting</li>
<li>Focus on genes reproducible across studies (increases effective n)</li>
<li>Report confidence intervals, not just point estimates</li>
</ul>
<p><strong>In this project:</strong> Combined datasets (e.g., <a href="https://github.com/Resilience-Biomarkers-for-Aquaculture/Cvirg_Pmarinus_RNAseq/issues/34">Issue #34</a> merged study 4 + 5) to increase sample size</p>
<h3 id="6-innate-vs-reactive-biomarkers">6. Innate vs. Reactive Biomarkers</h3>
<div class="admonition success">
<p class="admonition-title">Big Lesson #3: Biomarkers in Controls</p>
<p>Biomarkers may be <strong>constitutively expressed</strong> (innate) in resistant vs. sensitive individuals, even in the absence of stress.</p>
<p><strong>Implication:</strong> If you filter based on "stress-responsive" (control vs. treated), you will remove innate biomarkers.</p>
</div>
<p><strong>Detection:</strong></p>
<ul>
<li>Compare resistant vs. sensitive in <strong>control samples only</strong></li>
<li>If significant → innate biomarker</li>
<li>If not significant in controls but significant in treated → reactive biomarker</li>
</ul>
<p><strong>Pipeline choice:</strong></p>
<ul>
<li>✅ <strong>Two-step classifier</strong> preserves innate biomarkers</li>
<li>❌ <strong>Stepwise approach</strong> removes innate biomarkers by design</li>
</ul>
<p><strong>Related:</strong></p>
<ul>
<li><a href="https://github.com/Resilience-Biomarkers-for-Aquaculture/Cvirg_Pmarinus_RNAseq/issues/53">Issue #53: Innate vs. reactive analysis</a></li>
<li><a href="https://resilience-biomarkers-for-aquaculture.github.io/SY-innate-gene-expression/">Notebook: Innate gene expression</a></li>
</ul>
<hr />
<h2 id="validation-checklist">Validation Checklist</h2>
<p>Before claiming a validated biomarker panel:</p>
<ul>
<li>[ ] <strong>Cross-study validation:</strong> LOSO accuracy &gt; 70% (ideally &gt; 80%)</li>
<li>[ ] <strong>No data leakage:</strong> Normalization, feature selection, and hyperparameter tuning done independently per fold</li>
<li>[ ] <strong>Batch effect assessment:</strong> Confirmed genes are reproducible across studies, not driven by batch</li>
<li>[ ] <strong>Sample size adequacy:</strong> Each phenotype group has n &gt; 10 per study (if possible)</li>
<li>[ ] <strong>Phenotype definition:</strong> Clear, specific, and documented; comparable across studies used</li>
<li>[ ] <strong>Innate/reactive characterization:</strong> Assessed whether biomarkers are pre-existing or induced</li>
<li>[ ] <strong>Biological validation:</strong> Genes have known or plausible functional roles in stress response</li>
</ul>
<hr />
<h2 id="troubleshooting-guide">Troubleshooting Guide</h2>
<h3 id="poor-loso-performance">Poor LOSO Performance</h3>
<p><strong>Symptom:</strong> Accuracy &lt; 0.70 or high variance across folds</p>
<p><strong>Possible causes:</strong></p>
<ol>
<li><strong>Study-specific effects too strong</strong> → Filter datasets or use more stringent reproducibility criteria in Step 1</li>
<li><strong>Phenotype heterogeneity</strong> → Ensure comparable phenotype definitions</li>
<li><strong>Small sample sizes</strong> → Combine compatible studies or seek additional data</li>
<li><strong>Overfitting</strong> → Increase regularization penalty or reduce feature set</li>
</ol>
<h3 id="genes-dont-validate">Genes Don't Validate</h3>
<p><strong>Symptom:</strong> DEGs from discovery cohort not significant in validation cohort</p>
<p><strong>Possible causes:</strong></p>
<ol>
<li><strong>Batch effects</strong> → Apply batch-aware normalization or use post-data integration</li>
<li><strong>False discoveries</strong> → Discovery set had high FDR; use more stringent thresholds</li>
<li><strong>Different experimental conditions</strong> → Validation study not truly comparable</li>
<li><strong>Underpowered validation</strong> → Validation cohort too small to detect effects</li>
</ol>
<h3 id="suspect-data-leakage">Suspect Data Leakage</h3>
<p><strong>Symptom:</strong> Training accuracy much lower than test accuracy, or perfect test accuracy</p>
<p><strong>Diagnostic steps:</strong></p>
<ol>
<li>Re-run analysis with explicit train/test separation at each step</li>
<li>Check if normalization used all data before splitting</li>
<li>Verify feature selection used only training data</li>
<li>Ensure no test samples used in hyperparameter tuning</li>
</ol>
<hr />
<h2 id="related-resources">Related Resources</h2>
<p><strong>Key issues:</strong></p>
<ul>
<li><a href="https://github.com/Resilience-Biomarkers-for-Aquaculture/Cvirg_Pmarinus_RNAseq/issues/18">#18: Batch correction attempts</a></li>
<li><a href="https://github.com/Resilience-Biomarkers-for-Aquaculture/Cvirg_Pmarinus_RNAseq/issues/34">#34: Combining studies</a></li>
<li><a href="https://github.com/Resilience-Biomarkers-for-Aquaculture/Cvirg_Pmarinus_RNAseq/issues/44">#44: Classifier development with LOSO</a></li>
<li><a href="https://github.com/Resilience-Biomarkers-for-Aquaculture/Cvirg_Pmarinus_RNAseq/issues/53">#53: Innate vs. reactive</a></li>
</ul>
<p><strong>Key notebooks:</strong></p>
<ul>
<li><a href="https://resilience-biomarkers-for-aquaculture.github.io/SY-gene-classifier-panel/">Gene classifier panel (LOSO validation)</a></li>
<li><a href="https://resilience-biomarkers-for-aquaculture.github.io/SY-innate-gene-expression/">Innate gene expression analysis</a></li>
<li><a href="https://resilience-biomarkers-for-aquaculture.github.io/SY-six-gene-biomarker-exploration/">Six-gene biomarker exploration</a></li>
</ul>
<hr />
<p><strong>Next:</strong> Explore detailed project history in the <a href="../../timeline/">Timeline</a> or browse <a href="../../sources/issues-index/">Source Indices</a></p>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        
<div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/Resilience-Biomarkers-for-Aquaculture" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      
      <script id="__config" type="application/json">{"annotate": null, "base": "../..", "features": ["navigation.instant", "navigation.tracking", "navigation.tabs", "navigation.sections", "navigation.expand", "navigation.top", "search.suggest", "search.highlight", "content.code.copy"], "search": "../../assets/javascripts/workers/search.2c215733.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.79ae519e.min.js"></script>
      
    
  </body>
</html>