---
layout: post
title: Wed. Nov 14, 2018 Meeting with Kaitlyn about Oyster seed Proteomics paper
---

Proteomics manuscript next steps:

1. compare ASCA proteins with uniquely clustered proteins (Kaitlyn will look into this)
	- what do their abundances look like?
		- make raw abundance line plots facetted by protein
	- does the GO enrichment change when ASCA and cluster proteins are combined?

2. does hierarchical clustering change when day 0 is removed? (Kaitlyn will try this)

3. identify and locate data files that we need to deposit in public protein repository (i.e. ProteomeXchange and PeptideAtlas) (I will start this)
	- see what Emma's paper (and supplementary materials) included: 
		Raw data can be accessed via ProteomeXchange under
		identifier PXD004921. Raw data can be accessed in the
		PeptideAtlas under accession PASS00943 and PASS00942.
		Code used to perform enrichment analysis is available in a
		corresponding GitHub repository (https://github.com/
		yeastrc/compgo-geoduck-public) as is the underlying code for
		the end user web-interface (http://yeastrc.org/compgo_
		geoduck/pages/goAnalysisForm.jsp).
	- add file names and location, links to a googlesheet
	- confirm with Emma what files need to be made available
	- confirm with Rhonda where files are if we can't find them
	- do we include a fasta file of the peptides ID'd by mass spec?

4. add updated NMDS with only silo 3 and 9 to manuscript draft (Kaitlyn will do this)

5. For supplementary table of all proteins detected:
	- re-do the BLAST of CHOYP protein sequences to 2018 UniProt database (for reproducibility and to use up-to-date info)
	- ask Steven about files related to SQLShare from April 26, 2017
		- need to regenerate 'table_blastout_gigatonpep-uniprot'
		- need fasta file of the peptides ID'd by mass spec
		- need 2018 Uniprot DB
 	- make a simplified supplementary table containing CHOYP IDs, UniProt Accessions, e.val, Protein names, Gene names

6. do we need to explain we did 2 x 4 treatments, or just say we did 1 x 2 treatments?
	- do we have survival data for other silos to compare to silo 2, 3, and 9? 
	- Can we rule out silo 2 as an anomoly or should we include it?

7. re-focus the intro

----
****