---
layout: post
title: Mon. Oct 22, 2018
---

- updated author list and re-approved Sci Reports manuscript

- Corrected Pgenr Mox script to run Bismark on subset of Holly's BS data, and submitted job (in queue because nodes are being used)

- Submitted Bismark job on full Cvirg dataset on Mox with analysis time extended to 10 days because the previous time (1d15h) was not enough to finish the job, hence why it timed out.

	- copied over incomplete analysis to Cvirg metacarcinus folder on Gannet to try methylkit analysis while waiting for job to complete:
		pwd 
		/Volumes/web/metacarcinus/Cvirginica
		
		rsync --archive --progress --verbose strigg@mox.hyak.uw.edu:/gscratch/srlab/strigg/analyses/20181004 .

- lab meeting @1pm

- ELISA detection of vitellogenin in Geoduck:  
	- emailed sgarcia@uabc.edu.mx on 10-12-2018 about C. corteziensis Vtg/Vn polyclonal antibody from Fabiolaâ€™s paper
		- Fabiola's paper: compared translated mRNA sequences of Vtg/Vn from Panopea globosa and Crassostrea corteziensis to see if C. corteziensis VTG/Vn antibody would be compatible. I wanted to see the percent mapping between these protein sequences to get a sense of how much alignment and what domain homology is needed, but I couldn't find these sequences in the Genbank as the paper says.

	- Contacted TECO about their [multispecies ELISA kit for vitellogenin detection in fish](https://github.com/shellytrigg/P_generosa/raw/master/Vitellogenin_alignments/TECO-Vitellogenin-System-30.05.17EN.pdf):
Elizabeth Mihal @ TECO said (Oct 11, 2018): "Unfortunately, we do not have any indication that the Multispecies VTG kit would cross react with shellfish. I have attached a [cross reactivity chart](https://github.com/shellytrigg/P_generosa/raw/master/Vitellogenin_alignments/ML-00-00247REV01%20TECO_Vitellogenin_ELISA_Crossreactiviy_Chart.pdf) for your reference where you can see what species of fish have been shown to cross react with each kit. It is my understanding that the VTG in shellfish is too divergent from the VTG of Japanese rice fish, Australian rainbowfish, cod and flounder so it would make for an inappropriate sample type." Some info testing the kit compatibility with various fish species has [been published](https://setac.onlinelibrary.wiley.com/doi/full/10.1002/etc.3475)

	- tried general protein alignment of Geoduck VTG with BLASTP
		- made [fasta file](https://github.com/shellytrigg/P_generosa/blob/master/Vitellogenin_alignments/Vitellogenin_Pgenerosa.fa) with Geoduck vitellogenin sequence from Emma's paper (cds.comp144315_c0_seqI|m50928) 
		- performed BLASTP analysis using [BLAST website](https://blast.ncbi.nlm.nih.gov/Blast.cgi)
			- alignment graphical summary
				![alignment graphical summary](https://raw.githubusercontent.com/shellytrigg/P_generosa/master/Vitellogenin_alignments/BlastP-PgenrVTG-alignments.png)
			- [bowser view of alignments](https://goo.gl/h7Gp4J)
			- top alignments
				![top alignments](https://raw.githubusercontent.com/shellytrigg/P_generosa/master/Vitellogenin_alignments/BlastP-PgenrVTG-alignment-names.png)
			- top hit
				![top hit](https://raw.githubusercontent.com/shellytrigg/P_generosa/master/Vitellogenin_alignments/BlastP-PgenrVTG-tophit.png)
			- [Organism summary](https://github.com/shellytrigg/P_generosa/blob/master/Vitellogenin_alignments/NCBI%20Blast-ProteinSequenceOrganismSummary(2501%20letters).pdf) 
			- Phylogenic tree
				![Phylogenic Tree](https://raw.githubusercontent.com/shellytrigg/P_generosa/master/Vitellogenin_alignments/phylogenicTree_PgenrVTG_BLASTp.png)
		
		**Conclusions:** Pgenr VTG aligns best with VTG from other bivalves; BLASTp returned no species listed in TECO kits
	
	- tried protein alignment of Geoduck VTG to VTG from species used in TECO kit:
		- first tried aligning first Geoduck VTG domain which BLAST identified as 'Vitellogenin N superfamily' to species specific data by entering organism in search set options, figuring this domain was most conserved.
			- [alignments to zebrafish (which TECO kit works on)](https://github.com/shellytrigg/P_generosa/blob/master/Vitellogenin_alignments/Pgenr32-687_Daniorerio-Alignment.txt)
			- [alignments to Japanese rice fish](https://github.com/shellytrigg/P_generosa/blob/master/Vitellogenin_alignments/Pgenr32-687_Oryziaslatipes-Alignment.txt)
			- I compared these alignments to [the alignment of two fish species (zebrafish and goldfish)](https://github.com/shellytrigg/P_generosa/blob/master/Vitellogenin_alignments/GoldFishVTG1_Daniorerio-Alignment.txt) that show medium crossreactivity with the ELISA kit 
		
		- next made [fasta file](https://github.com/shellytrigg/P_generosa/blob/master/Vitellogenin_alignments/MultiSpecies_VTG.fa) with different species with detectable VTG in TECO kit
		- used BLASTp with option to align multiple sequences and uploaded [the Pgenr fasta file](https://github.com/shellytrigg/P_generosa/blob/master/Vitellogenin_alignments/Vitellogenin_Pgenerosa.fa) and [the TECO species fasta file](https://github.com/shellytrigg/P_generosa/blob/master/Vitellogenin_alignments/MultiSpecies_VTG.fa)
			- Phylogenic tree
				![Phylogenic tree](https://raw.githubusercontent.com/shellytrigg/P_generosa/master/Vitellogenin_alignments/PhylogenicTreeTECOorgsVTGseqs.png)
			- [alignments](https://www.ebi.ac.uk/Tools/services/rest/mview/result/mview-I20181023-021116-0380-96603652-p1m/aln-html)
		-used [clustal omega](https://www.ebi.ac.uk/Tools/msa/clustalo/) to perform multiple alignment with default settings
			- [alignments](https://github.com/shellytrigg/P_generosa/blob/master/Vitellogenin_alignments/ClustalOmega.pdf)
	

	**Conclusions:** TECO species VTGs align much better to each other than Pgenr VTG aligns with TECO species VTG. Pgenr VTG is more similar to C. gigas than to any TECO fish species and Pgenr VTG is most likely too divergent from TECO fish species. 

	mRNA expression may be the better way to go

	
----
****