---
layout: post
title: Mon. Nov 26, 2018
---


9am-10am:  Picked up Floy tags


## Geoduck Juvenile OA RRBS

- Started running methylkit on Ostrich on all juvenile methylation data
	- takes about 25 min per dedup.sorted.bam file for methylkit to read it in

- How does R studio desktop on Ostrich (running 1 core @ 100% CPU) compare to Emu R studio server?

	1. log in to R studio server:  http://emu.fish.washington.edu:8787/

	2. Making zip file of dedup bams that can be downloaded to local computer:
	
	```
		mkdir dedup_sorted_bam
		cp *dedup.sorted.bam dedup_sorted_bam/
		tar -zcvf dedup_sorted_bam.tar.gz dedup_sorted_bam
			a dedup_sorted_bam
			a dedup_sorted_bam/EPI-162_S9_L002_dedup.sorted.bam
			a dedup_sorted_bam/EPI-42_S39_L005_dedup.sorted.bam
			a dedup_sorted_bam/EPI-176_S15_L003_dedup.sorted.bam
			a dedup_sorted_bam/EPI-214_S30_L004_dedup.sorted.bam
			a dedup_sorted_bam/EPI-135_S35_L005_dedup.sorted.bam
			a dedup_sorted_bam/EPI-170_S13_L002_dedup.sorted.bam
			a dedup_sorted_bam/EPI-113_S30_L005_dedup.sorted.bam
			a dedup_sorted_bam/EPI-206_S27_L004_dedup.sorted.bam
			a dedup_sorted_bam/EPI-41_S38_L005_dedup.sorted.bam
			a dedup_sorted_bam/EPI-151_S2_L002_dedup.sorted.bam
			a dedup_sorted_bam/EPI-44_S41_L005_dedup.sorted.bam
			a dedup_sorted_bam/EPI-161_S8_L002_dedup.sorted.bam
			a dedup_sorted_bam/EPI-136_S36_L005_dedup.sorted.bam
			a dedup_sorted_bam/EPI-205_S26_L004_dedup.sorted.bam
			a dedup_sorted_bam/EPI-193_S22_L003_dedup.sorted.bam
			a dedup_sorted_bam/EPI-182_S17_L003_dedup.sorted.bam
			a dedup_sorted_bam/EPI-226_S34_L004_dedup.sorted.bam
			a dedup_sorted_bam/EPI-229_S36_L004_dedup.sorted.bam
			a dedup_sorted_bam/EPI-168_S11_L002_dedup.sorted.bam
			a dedup_sorted_bam/EPI-220_S32_L004_dedup.sorted.bam
			a dedup_sorted_bam/EPI-145_S38_L005_dedup.sorted.bam
			a dedup_sorted_bam/EPI-167_S10_L002_dedup.sorted.bam
			a dedup_sorted_bam/EPI-103_S27_L005_dedup.sorted.bam
			a dedup_sorted_bam/EPI-215_S31_L004_dedup.sorted.bam
			a dedup_sorted_bam/EPI-208_S28_L004_dedup.sorted.bam
			a dedup_sorted_bam/EPI-119_S31_L005_dedup.sorted.bam
			a dedup_sorted_bam/EPI-159_S6_L002_dedup.sorted.bam
			a dedup_sorted_bam/EPI-227_S35_L004_dedup.sorted.bam
			a dedup_sorted_bam/EPI-209_S29_L004_dedup.sorted.bam
			a dedup_sorted_bam/EPI-153_S4_L002_dedup.sorted.bam
			a dedup_sorted_bam/EPI-181_S16_L003_dedup.sorted.bam
			a dedup_sorted_bam/EPI-199_S24_L003_dedup.sorted.bam
			a dedup_sorted_bam/EPI-120_S32_L005_dedup.sorted.bam
			a dedup_sorted_bam/EPI-160_S7_L002_dedup.sorted.bam
			a dedup_sorted_bam/EPI-221_S33_L004_dedup.sorted.bam
			a dedup_sorted_bam/EPI-230_S37_L004_dedup.sorted.bam
			a dedup_sorted_bam/EPI-143_S37_L005_dedup.sorted.bam
			a dedup_sorted_bam/EPI-128_S34_L005_dedup.sorted.bam
			a dedup_sorted_bam/EPI-184_S18_L003_dedup.sorted.bam
			a dedup_sorted_bam/EPI-200_S25_L003_dedup.sorted.bam
			a dedup_sorted_bam/EPI-185_S19_L003_dedup.sorted.bam
			a dedup_sorted_bam/EPI-175_S14_L003_dedup.sorted.bam
			a dedup_sorted_bam/EPI-194_S23_L003_dedup.sorted.bam
			a dedup_sorted_bam/EPI-187_S20_L003_dedup.sorted.bam
			a dedup_sorted_bam/EPI-104_S28_L005_dedup.sorted.bam
			a dedup_sorted_bam/EPI-188_S21_L003_dedup.sorted.bam
			a dedup_sorted_bam/EPI-111_S29_L005_dedup.sorted.bam
			a dedup_sorted_bam/EPI-169_S12_L002_dedup.sorted.bam
			a dedup_sorted_bam/EPI-127_S33_L005_dedup.sorted.bam
			a dedup_sorted_bam/EPI-154_S5_L002_dedup.sorted.bam
			a dedup_sorted_bam/EPI-43_S40_L005_dedup.sorted.bam
			a dedup_sorted_bam/EPI-152_S3_L002_dedup.sorted.bam
	```

	3. downloaded and unzipped tar in R


## Calculating genome coverage with Bedtools
1. Make genome coordinates file with chromosome names consistent with as they appear in the fasta and in the .bam file
2. coverageBed

	```
	Default Output:  
	 After each entry in A, reports: 
	   1) The number of features in B that overlapped the A interval.
	   2) The number of bases in A that had non-zero coverage.
	   3) The length of the entry in A.
	   4) The fraction of bases in A that had non-zero coverage
	```
	
```
/Users/Shared/bioinformatics/bedtools2/bin/coverageBed -a EPI-168_S11_L002_dedup.sorted.bam -b /Volumes/web/metacarcinus/Pgenerosa/GENOMES/v070/Pgenerosa_NW100.bed| head
PGA_scaffold1__77_contigs__length_89643857	499267	499368	D00743:144:CAAWNANXX:2:2207:10523:3493_1:N:0:TAGCTT/1	6	+	499267	499368	0,0,0	1	101,	0,	0	0	101	0.0000000
PGA_scaffold1__77_contigs__length_89643857	499275	499374	D00743:144:CAAWNANXX:2:2207:10523:3493_1:N:0:TAGCTT/2	6	-	499275	499374	0,0,0	1	99,	0,	0	0	99	0.0000000
PGA_scaffold1__77_contigs__length_89643857	1421232	1421329	D00743:144:CAAWNANXX:2:2207:12810:4640_1:N:0:TAGCTT/2	6	+	1421232	1421329	0,0,0	1	97,	0,	0	0	97	0.0000000
PGA_scaffold1__77_contigs__length_89643857	1421316	1421417	D00743:144:CAAWNANXX:2:2207:12810:4640_1:N:0:TAGCTT/1	6	-	1421316	1421417	0,0,0	1	101,	0,	0	0	101	0.0000000
PGA_scaffold1__77_contigs__length_89643857	1482262	1482361	D00743:144:CAAWNANXX:2:2207:20644:5525_1:N:0:TAGCTT/2	42	+	1482262	1482361	0,0,0	1	99,	0,	0	0	99	0.0000000
PGA_scaffold1__77_contigs__length_89643857	1482501	1482602	D00743:144:CAAWNANXX:2:2207:20644:5525_1:N:0:TAGCTT/1	42	-	1482501	1482602	0,0,0	1	101,	0,	0	0	101	0.0000000
PGA_scaffold1__77_contigs__length_89643857	1897695	1897785	D00743:144:CAAWNANXX:2:2207:9889:3016_1:N:0:TAGCTT/2	6	+	1897695	1897785	0,0,0	1	90,	0,	0	0	90	0.0000000
PGA_scaffold1__77_contigs__length_89643857	1897695	1897785	D00743:144:CAAWNANXX:2:2207:9889:3016_1:N:0:TAGCTT/1	6	-	1897695	1897785	0,0,0	1	90,	0,	0	0	90	0.0000000
PGA_scaffold1__77_contigs__length_89643857	2101090	2101187	D00743:144:CAAWNANXX:2:2207:15137:5638_1:N:0:TAGCTT/2	0	+	2101090	2101187	0,0,0	1	97,	0,	0	0	97	0.0000000
PGA_scaffold1__77_contigs__length_89643857	2101101	2101165	D00743:144:CAAWNANXX:2:2207:15137:5638_1:N:0:TAGCTT/1	0	-	2101101	2101165	0,0,0	1	64,	0,	0	0	64	0.0000000
```

```

/Users/Shared/bioinformatics/bedtools2/bin/coverageBed -a /Volumes/web/metacarcinus/Pgenerosa/20181101/EPI-168_S11_L002_dedup.sorted.bam -b /Volumes/web/metacarcinus/Pgenerosa/GENOMES/v070/Pgenerosa_W10000.bed| head
PGA_scaffold1__77_contigs__length_89643857	0	93	D00743:144:CAAWNANXX:2:2301:1127:5871_1:N:0:TAGCTT/1	40	+	0	93	0,0,0	1	93,	0,	1	93	93	1.0000000
PGA_scaffold1__77_contigs__length_89643857	6	105	D00743:144:CAAWNANXX:2:2301:1127:5871_1:N:0:TAGCTT/2	40	-	6	105	0,0,0	1	99,	0,	1	99	99	1.0000000
PGA_scaffold1__77_contigs__length_89643857	181	282	D00743:144:CAAWNANXX:2:1204:2228:71694_1:N:0:TAGCTT/1	40	+	181	282	0,0,0	1	101,	0,	1	101	101	1.0000000
PGA_scaffold1__77_contigs__length_89643857	521	611	D00743:144:CAAWNANXX:2:1204:2228:71694_1:N:0:TAGCTT/2	40	-	521	611	0,0,0	1	90,	0,	1	90	90	1.0000000
PGA_scaffold1__77_contigs__length_89643857	535	632	D00743:144:CAAWNANXX:2:2203:16222:89089_1:N:0:TAGCTT/1	3	+	535	632	0,0,0	1	97,	0,	1	97	97	1.0000000
PGA_scaffold1__77_contigs__length_89643857	577	662	D00743:144:CAAWNANXX:2:2203:16222:89089_1:N:0:TAGCTT/2	3	-	577	662	0,0,0	1	85,	0,	1	85	85	1.0000000
PGA_scaffold1__77_contigs__length_89643857	1867	1965	D00743:144:CAAWNANXX:2:2114:5007:16095_1:N:0:TAGCTT/2	42	+	1867	1965	0,0,0	1	98,	0,	1	98	98	1.0000000
PGA_scaffold1__77_contigs__length_89643857	1881	1982	D00743:144:CAAWNANXX:2:1304:3756:67096_1:N:0:TAGCTT/1	23	+	1881	1982	0,0,0	1	101,	0,	1	101	101	1.0000000
PGA_scaffold1__77_contigs__length_89643857	1902	2000	D00743:144:CAAWNANXX:2:2114:5007:16095_1:N:0:TAGCTT/1	42	-	1902	2000	0,0,0	1	98,	0,	1	98	98	1.0000000
PGA_scaffold1__77_contigs__length_89643857	1998	2097	D00743:144:CAAWNANXX:2:1311:4932:14375_1:N:0:TAGCTT/2	42	+	1998	2097	0,0,0	1	99,	0,	1	99	99	1.0000000
```

```
/Users/Shared/bioinformatics/bedtools2/bin/coverageBed -a /Volumes/web/metacarcinus/Pgenerosa/GENOMES/v070/Pgenerosa_W10000.bed -b /Volumes/web/metacarcinus/Pgenerosa/20181101/EPI-168_S11_L002_dedup.sorted.bam| head
PGA_scaffold1__77_contigs__length_89643857	0	10000	112	3889	10000	0.3889000
PGA_scaffold1__77_contigs__length_89643857	10000	20000	74	3297	10000	0.3297000
PGA_scaffold1__77_contigs__length_89643857	20000	30000	109	4113	10000	0.4113000
PGA_scaffold1__77_contigs__length_89643857	30000	40000	135	4122	10000	0.4122000
PGA_scaffold1__77_contigs__length_89643857	40000	50000	112	3955	10000	0.3955000
PGA_scaffold1__77_contigs__length_89643857	50000	60000	94	3425	10000	0.3425000
PGA_scaffold1__77_contigs__length_89643857	60000	70000	100	4310	10000	0.4310000
PGA_scaffold1__77_contigs__length_89643857	70000	80000	281	6625	10000	0.6625000
PGA_scaffold1__77_contigs__length_89643857	80000	90000	258	6742	10000	0.6742000
PGA_scaffold1__77_contigs__length_89643857	90000	100000	108	4732	10000	0.4732000
```

## GenomeCov
- This does not give even bin sizes so will be hard to compare across samples
    1. chromosome (or entire genome)
    2. depth of coverage from features in input file
    3. number of bases on chromosome (or genome) with depth equal to column 2.
    4. size of chromosome (or entire genome) in base pairs
    5. fraction of bases on chromosome (or entire genome) with depth equal to column 2.


```
/Users/Shared/bioinformatics/bedtools2/bin/genomeCoverageBed -ibam /Volumes/web/metacarcinus/Pgenerosa/20181101/EPI-168_S11_L002_dedup.sorted.bam -g /Volumes/web/metacarcinus/Pgenerosa/GENOMES/v070/Pgenerosa_v070_chrm_names_and_lengths.txt > /Volumes/web/metacarcinus/Pgenerosa/20181101/EPI-168_S11_L002_dedup.sorted.genomecov

PGA_scaffold2__36_contigs__length_69596280      0       43054322        69596280        0.61863
PGA_scaffold2__36_contigs__length_69596280      1       9827263 69596280        0.141204
PGA_scaffold2__36_contigs__length_69596280      2       6414167 69596280        0.0921625
PGA_scaffold2__36_contigs__length_69596280      3       3170774 69596280        0.0455595
PGA_scaffold2__36_contigs__length_69596280      4       2228630 69596280        0.0320223
PGA_scaffold2__36_contigs__length_69596280      5       1332827 69596280        0.0191508
PGA_scaffold2__36_contigs__length_69596280      6       970855  69596280        0.0139498
```
