---
layout: post
title: Fri. May 31, 2019  Methyl CpGoe analysis  
---

### This post is related to [github issue #694](https://github.com/RobertsLab/resources/issues/694)

The main problem was not being able to combine the ID\_CpG\_labelled outputs together into one file as Sam was able to do in his [original CpGoe analysis](https://gannet.fish.washington.edu/Atumefaciens/20190225_cpg_oe/) using this [script](https://gannet.fish.washington.edu/Atumefaciens/20190225_cpg_oe/20190226_join_ID_CpGs.sh).

Yaamini created new CpGoe files specific for CAP, CDS, GENE, Exons, and windows features of the genome mentioned in [this post](https://genefish.wordpress.com/2019/05/28/yaaminis-notebook-froger-recap/) and following [this markdown file](https://github.com/hputnam/FROGER/blob/master/analyses/CpGoe/20190521_CpGOE_methods.md). After having problems with the script adding ID_CpG file headers and joining the files together, I attempted to run this script 

**on Mox:**

1. First I copied the data and script over to Mox:
 	- mkdir /gscratch/srlab/strigg/data/Cvirg/FROGER\_CAP\_CpGoe
 	- cd /gscratch/srlab/strigg/data/Cvirg/FROGER\_CAP\_CpGoe
	- rsync --archive --progress --verbose yaamini@172.25.149.226:/Volumes/web/spartina/2019-05-21-FROGER/CAP_CpGoe .

2. Next I turned this script (/gscratch/srlab/strigg/data/Cvirg/FROGER\_CAP\_CpGoe/CAP\_CpGoe/2019-05-22-CAP-File-Naming-Script.sh) into a Mox job (/gscratch/srlab/strigg/jobs/20190523\_CAP\_File\_Naming.sh) and ran it on May 23. It errored out in the join step after 3-4 hours of running and joining 26 of the 90 files. 

	- error message: "Broken pipe join --nocheck-order ID\_CpG\_labelled\_all ${file}ID\_CpG\_labelled "
	- slurm file is here:
/gscratch/srlab/strigg/data/Cvirg/FROGER\_CAP\_CpGoe/CAP\_CpGoe/slurm-863764.out

3. I copied everything to my folder on Gannet [https://gannet.fish.washington.edu/metacarcinus/Cvirginica/FROGER/](https://gannet.fish.washington.edu/metacarcinus/Cvirginica/FROGER/)

4. I realized some of the sample names have more than one underscore and having a non-unique column name could have been problematic.

5. We also learned that certain bash commands (sed and maybe even the join command) may not run the same way on a mac (how Yaamini and I have tried running the script) as they do on a PC (how Sam ran it)

6. I made changes to the script to 
	- handle any number of underscores in the sample names
	- simplify adding the header with the cat command instead of the sed command
	- using paste and awk commands in a one-liner to join the files instead of a loop with the join command 
	- see [updated script https://gannet.fish.washington.edu/metacarcinus/Cvirginica/FROGER/CAP_CpGoe/20190531_append_headers_join_ID_CpGs.sh](https://gannet.fish.washington.edu/metacarcinus/Cvirginica/FROGER/CAP_CpGoe/20190531_append_headers_join_ID_CpGs.sh) for explanation of changes and new code

**On Ostrich:**

1. I used [this jupyter notebook](https://github.com/hputnam/FROGER/blob/master/analyses/CAP_CpGoe/CpGoe_append_headers_join_files.ipynb) to 
	- run the updated script on the CAP data 
	- run the updated script on the GENE data to duplicate what Sam already did so I could compare the output of the new script to Sam's.

2. **QC**: In R, I compared [output of new script](https://gannet.fish.washington.edu/metacarcinus/Cvirginica/FROGER/20190225_cpg_oe/ID_CpG_labelled_all) with [output of Sam's orginial script that I reformatted](https://gannet.fish.washington.edu/metacarcinus/Cvirginica/FROGER/20190225_cpg_oe/ID_CpG_labelled_all.tab) in R. 
	- See [the output](http://htmlpreview.github.io/?https://github.com/hputnam/FROGER/blob/master/analyses/CAP_CpGoe/CpGoe_append_headers_join_files_QC.html) of [this markdown file](https://github.com/hputnam/FROGER/blob/master/analyses/CAP_CpGoe/CpGoe_append_headers_join_files_QC.Rmd) for R analysis.

### CONCLUSIONS:
- [updated script](https://gannet.fish.washington.edu/metacarcinus/Cvirginica/FROGER/CAP_CpGoe/20190531_append_headers_join_ID_CpGs.sh) gives same output as original script and can now be used on remaining CpGoe analyses (CDS, Exons, windows).  


