---
layout: post
title: Tues. Dec. 17, DMR pipeline 
---

STEP 1: Bismark alignments
A. Build bismark genome
B. Align FASTQs
C. Sort bam files


STEP 2: Methylpy
A. Call Methylation state
	- mox script: 
	- allc files filtered for MAPQ20
B. Calculate total methylation in experimental groups
	- mox script: FiltCombIntscAllc
		- this script filters allc files for 5x coverage
		- data: allc_5x_CpG.txt
	- jupyter: CalculateGlobalMethylation_5x0.75group_CpGs.ipynb
	- R script: CpGoverall_stats.Rmd
C. Run DMRfind
	- mox script:
D. Filter DMRs for coverage in at least 3/4 individuals/group
	- R script: DMR_cov_in_0.75_SamplesPerCategory.Rmd
E. Run group statistics
	- R script: MCmax25_asinT_groupStats.Rmd

STEP 3: GO analysis
A. Map to uniprot DB to get GO terms
B. Run TopGO to do GO enrichment on DMRs
C. For genes with enriched GO terms, convert GO terms to GO slim with GSEA base
C. Plot # proteins/GO slim