---
layout: post
title: Tues. Jul. 23, 2019  Salmon + sea lice methylomes and Oyster Proteomics 
---

## Oyster Proteomics

- Cleaned up the [manuscript](https://docs.google.com/document/d/1OusRSi9W5qLRcc3nuJm-kyQYrTQb7yPFiCBtqZIKNpQ/edit)
	- still think discussion and interpretation need work
		- would be good (and probably would be requested by reviewer) to compare to other proteomics results
			- found 3 papers that would make cool comparisons:
				1. [C.gigas shell-matrix proteome](https://www.nature.com/articles/srep45754) 
				2. [C.gigas thermal stress proteome](https://www.ncbi.nlm.nih.gov/pubmed/26990129)
				3. [Viral response proteome](https://www.sciencedirect.com/science/article/pii/S1874391914003455?via%3Dihub#t0010)
					- [dsRNA response proteome](https://pubs.acs.org/doi/abs/10.1021/acs.jproteome.5b00615?src=recsys)

					
## Salmon + sea lice methylomes  
- Still running TrimGalore! (probably will be around 30 hours to complete) 
- Prepared Bismark genomes on Mox:
	- Salmon genome prep
		- script location:  /gscratch/srlab/strigg/jobs/BuildSalmo_BmrkGenome.sh
		- bismark genome location:  /gscratch/srlab/strigg/data/Ssalar/GENOMES
	- Sea lice genome prep
		- script location:  /gscratch/srlab/strigg/jobs/BuildCalig_BmrkGenome.sh
		- bismark genome location:  /gscratch/srlab/strigg/data/Caligus/GENOMES
- Determine bismark alignment settings to use
	- **Revisit past analysis:**
		- I went back and did a quick but more thorough comparison
		- **RESULTS** here: [https://github.com/shellywanamaker/C_virginica/blob/master/AlignementScoreComparison/BmrkAlignementScoreComparison.html](http://htmlpreview.github.io/?https://github.com/shellywanamaker/C_virginica/blob/master/AlignementScoreComparison/BmrkAlignementScoreComparison.html)
		- BmrkAlignementScoreComparison.Rmd code here: [https://github.com/shellywanamaker/C_virginica/tree/master/AlignementScoreComparison](https://github.com/shellywanamaker/C_virginica/tree/master/AlignementScoreComparison)
		- **CONCLUSIONS**: 
			- I still think in this case AS 0,-1.2 -I 60 was the best compromise because it gives results closest to AS 0,-0.6 with a greater mapping effieciency (NOTE: mapping effieciency is the % of sequence pairs with unique alignments) and not much increase in artifical inflation of %C's in CHH, CN or CHN, or CHG context. 
			- I'm still not totally sure what we should be expecting for %C's in CHH, CN or CHN, CHG, or CpG context, so that makes this whole analysis hard to wrap my head around. 
		- **NEXT STEPS:** 
			- I will try different parameters for the salmon/sea lice samples:
				- AS 0, -0.2
				- AS 0, -0.6
				- AS 0, -1
				- AS 0, -1 -I60
				- AS 0, -1.2 -I60
				- AS 0, -2
			- [MOX scripts here:](https://github.com/shellywanamaker/Salmon_sealice/tree/master/Mox_Scripts)
				- [Bmrk\_CompareAS\_Calig_100K.sh](https://github.com/shellywanamaker/Salmon_sealice/blob/master/Mox_Scripts/BismarkAlign_Calig.sh)
				- [Bmrk\_CompareAS\_Salmo_100K.sh](https://github.com/shellywanamaker/Salmon_sealice/blob/master/Mox_Scripts/BismarkAlign_Ssalar.sh)
				- [BismarkAlign\_Calig.sh](https://github.com/shellywanamaker/Salmon_sealice/blob/master/Mox_Scripts/Bmrk_CompareAS_Calig_100K.sh)
				- [BismarkAlign\_Ssalar.sh](https://github.com/shellywanamaker/Salmon_sealice/blob/master/Mox_Scripts/Bmrk_CompareAS_Salmo_100K.sh)

			
		