---
layout: post
title: Aug. 13-14, 2019 
---

## Geoduck genome
- Review BamQC results
	- coverage plots over scaffold
	- compare coverage of:
		- v074 scaffolds 1-18
		- v070 scaffolds 1-18
		- v070 scaffolds 19-n
- Load DMRs and bams into IGV 	
- Write summary of overall analysis

## Sea lice and Salmon data
- Start full alignments using pre-determined settings on Mox
	- Created scripts:
		- Salmon:  [20190814_BmrkSsalar.sh](https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20190814_BmrkSsalar.sh)
		- Sea lice:  [20190814_BmrkCalig.sh](https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20190814_BmrkCalig.sh)  
	- Started running 20190814_BmrkSsalar.sh @ 11:40am
		- takes about 4.5-5 hours/sample so alignments should be done in ~4-6 days including rest of analysis (we'll see). 

- copy data from Ostrich desktop to Gannet
	- note: this is the same data as was copied to Mox /gscratch/scrubbed/strigg/TRIMG_adapt_5bp (see [this post](https://shellytrigg.github.io/146th-post/))

	```
 	rsync --archive --progress --verbose * /Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/ 

	rsync --archive --progress --verbose /gscratch/scrubbed/strigg/TRIMG_adapt_5bp/ strigg@ostrich.fish.washington.edu:/Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore


	```

- copy data from Mox to Gannet
	
	```
	rsync --archive --progress --verbose /gscratch/scrubbed/strigg/analyses/20190809_100K_Calig_trimG strigg@ostrich.fish.washington.edu:/Volumes/web/metacarcinus/Salmo_Calig/analyses/
	
	rsync --archive --progress --verbose /gscratch/scrubbed/strigg/analyses/20190809_100K_Salmo strigg@ostrich.fish.washington.edu:/Volumes/web/metacarcinus/Salmo_Calig/analyses/
	
	rsync --archive --progress --verbose /gscratch/scrubbed/strigg/analyses/20190809_100K_Salmo_trimME strigg@ostrich.fish.washington.edu:/Volumes/web/metacarcinus/Salmo_Calig/analyses/
	
	rsync --archive --progress --verbose /gscratch/srlab/strigg/jobs/ strigg@ostrich.fish.washington.edu:/Volumes/web/metacarcinus/mox_jobs/
	```
	

## Geoduck Broodstock conditioning manuscript
- Go through water chem (nov 2018- mar 2019)
	- Apex data:
		- 11/21 - 2/1 [here](https://github.com/shellytrigg/P_generosa/tree/master/Water_Chemistry/Data/Apex)
		- 2/1 - 3/17 **need to download**
	- Discrete measurments [here](https://github.com/shellytrigg/P_generosa/blob/master/Water_Chemistry/data/Titrator/Daily_Temp_pH_Sal.csv) 
	- Titrator:
		- 11/28 - 1/14 [here](https://github.com/shellytrigg/P_generosa/tree/master/Water_Chemistry/Data/Titrator)
		- 1/21 - 3/19 [here](https://github.com/SamGurr/Geoduck_transgen_offspring_OA/tree/master/RAnalysis/Data/TA)
	- **NEED TO:**  
		- Run cumulative R analysis of all titrator data
		- Consolidate all data into one table 
- Go through survival data of var. low pH animals
	- data [here](https://github.com/shellytrigg/P_generosa/blob/master/amb_v_varlow_brood/data/Tank5_6_Broodstock_survival_201811-20190317.xlsx)
	- analysis [here](https://github.com/shellytrigg/P_generosa/tree/master/amb_v_varlow_brood/analyses/survival) 
- Go through [respirometry data](https://github.com/shellytrigg/P_generosa/tree/master/amb_v_constantlow_brood_data/Broodstock_Respirometry) and see if it's usable
	- respirometry data is from [Broodstock exposed to constant low pH](https://github.com/shellytrigg/P_generosa/blob/master/2018-19_Brood_sample_timeline/pH_Constant.Low.png) 
- Write all in methods and results of manuscript
	- manuscript draft [here](https://docs.google.com/document/d/1aIho7R27-cXpDpuRKlQY_p-3Bi7-HlZMNQcTtH_bigI/edit) 
	- paper repo [here](https://github.com/shellytrigg/paper-GeoduckBrood_Juvs_pH)

## Oyster proteomics manuscript
- decide on journal
- continue comparing results to literature 
