---
layout: post
title: Tues. Oct. 1, 2019, Geoduck Juv. OA DMR analysis
---

### QC of DMRs
- created filtered bam files for doing quality check on the DMRs called from [9/26 script](https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20190926_DMRfind3clstAllEPIsamples.sh)
	- I only created filtered bam files for the all ambient timepoints comparison
	- jupyter notebook here (section noted as "This analysis below was done on 9/26/19") : [20190909_DMRallEPI_allc_minClst3.ipynb](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20190909_DMRallEPI_allc_minClst3.ipynb) 
	- **Filtered bam files for DMR QC are here:** [https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20190926/allAmb/](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20190926/allAmb/)
	- IGV session of filtered bam files is here: [https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20190926/allAmb/QC_20190926DMRs_allAmb.xml](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20190926/allAmb/QC_20190926DMRs_allAmb.xml)
		- DMR bed files in IGV:
			1. ["MAPQ30\_MCmax50\_cov5x_clst3"](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20190822/amb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_collapsed.tsv.DMR.bed) 
			2. ["MAPQ30\_MCmax25\_cov5x_clst3"](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20190822/amb_AllTimes_DMR250bp_MCmax25_cov5x_clst3_rms_results_collapsed.tsv.DMR.bed)
			3. ["MAPQ30\_MCmax25\_cov5x_clst2"](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20190822/amb_AllTimes_DMR250bp_MCmax25_cov5x_rms_results_collapsed.tsv.DMR.bed)
			4. ["MAPQ20\_MCmax25\_cov5x_clst3"](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20190926/amb_AllTimes_DMR250bp_MCmax25_cov5x_clst3_rms_results_collapsed.tsv.DMR.bed)
		- some notes about these files:
			- DMRs are commonly defined as 250bp regions containing minimally 3 differentially methylated sites (DMS) that have 5x coverage (noted in the track name as "cov5x")
			- MAPQ filter is specified at the beginning of the track name
			- MCmax25 refers to a parameter in DMRfind specifying that if DMS are within a 25bp window, their cytosine coverage can be summed. MCmax50 means the DMS can be within a 50bp window. 
			- "clst" in the track name refers to how many individuals within a treatment group need to have coverage of the DMR for it to be reported. "clst2" requires 2 individuals/group and "clst3" requires 3 individuals/group.
		- another DMR bed file that could be looked at in this session is from the analysis where data from all individuals in each group were merged so that each group had one allc file before DMRs were called. That file is [here](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20190822/merge_allc/amb_AllTimes_mergAllc_DMR250bp_MCmax25_cov5x_rms_results_collapsed.tsv.DMR.bed) 

### Functional analysis of DMRs
- created tables for DMRs from each of the 4 comparisons containing the following fields:
	1. chromosome (scaffold)
	2. start position
	3. end position
	4. number of DMS in DMR
	5. Gene ID
	6. Swissprot ID
	7. Swissprot Entry Name
	8. Protein name(s)
	9. GO ID
	10. GO term
- these tables are here:
	1. [amb_AllTimes.GO.txt](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20191001_anno/amb_AllTimes.GO.txt)
	2. [day10_AllpH.GO.txt](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20191001_anno/day10_AllpH.GO.txt)
	3. [day135_AllpH.GO.txt](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20191001_anno/day135_AllpH.GO.txt)
	4. [day145_AllpH.GO.txt](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20191001_anno/day145_AllpH.GO.txt)
- these tables were created with this jupyter notebook: [20191001_DMR_functional_analysis.ipynb](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20191001_DMR_functional_analysis.ipynb)
- I also created TE tables for each comparison, however there were not many TEs (because there are not that many DMRs in general)
	1. [amb\_AllTimes\_TE.txt](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20191001_anno/amb_AllTimes_TE.txt)
	2. [day10\_AllpH\_TE.txt](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20191001_anno/day10_AllpH_TE.txt)
	3. [day135\_AllpH\_TE.txt](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20191001_anno/day135_AllpH_TE.txt)
	4. [day145\_AllpH\_TE.txt](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20191001_anno/day145_AllpH_TE.txt)

### Next steps 
- Look into what proteins are and finish functional analysis
- see if genes identified overlap with Hollie's DMGs
- Decide if it's worth doing a domain enrichment analysis referred to in [this slack convo](https://genefish.slack.com/archives/GG4HW5SC9/p1569966253020700?thread_ts=1569347224.006000&cid=GG4HW5SC9)
- Do a more thorough QC check to make sure these DMRs are believable
	- Decide if any DMRfind parameters should be adjusted and rerun