---
layout: post
title: Sat. Nov. 2, Geoduck DMR analysis 
category: Geoduck epigenomics
---

## Group stats on transformed data

Because I can't easily generate the methylated and unmethylated C counts from which the percent methylation was generated (since I applied a parameter that summed counts from nearby bases and some other filtering parameters; see [DMRfind script](https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20191024_DMRfindAllEPI.sh)), Brent suggested transforming the data to make it more normal and then apply ANOVA. 

- To transform proportion data, you use arcsin(sqrt(%)). 
- I transformed the percent methylation DMR data in all 4 comparisons using this Rmarkdown script [MCmax30_asinT_groupStats.Rmd](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/MCmax30_asinT_groupStats/MCmax30_asinT_groupStats.Rmd) and Rproj [MCmax30_asinT_groupStats.Rproj](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/MCmax30_asinT_groupStats/MCmax30_asinT_groupStats.Rproj).
	- the percent methylation DMR data had already been filtered for those showing coverage across at least 3/4 individuals/group 
- Distribution of % methylation for each experimental group in each comparison before and after tranformation looked like this:
	- all ambient samples over time **BEFORE**:  [![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/allAmb_MCmax30DMR_percmeth_hist.jpg)](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/allAmb_MCmax30DMR_percmeth_hist.jpg)
	- all ambient samples over time **AFTER**:  [![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/allAmb_MCmax30DMR_Tpercmeth_hist.jpg)](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/allAmb_MCmax30DMR_Tpercmeth_hist.jpg)
- all day 10 samples **BEFORE**:  [![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day10_MCmax30DMR_percmeth_hist.jpg)](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day10_MCmax30DMR_percmeth_hist.jpg)
	- all day 10 samples **AFTER**:  [![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day10_MCmax30DMR_Tpercmeth_hist.jpg)](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day10_MCmax30DMR_Tpercmeth_hist.jpg)
- all day 135 samples **BEFORE**:  [![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day135_MCmax30DMR_percmeth_hist.jpg)](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day135_MCmax30DMR_percmeth_hist.jpg)
	- all day 135 samples **AFTER**:  [![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day135_MCmax30DMR_Tpercmeth_hist.jpg)](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day135_MCmax30DMR_Tpercmeth_hist.jpg)
- all day 145 samples **BEFORE**:  [![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day145_MCmax30DMR_percmeth_hist.jpg)](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day145_MCmax30DMR_percmeth_hist.jpg)
	- all day 145 samples **AFTER**:  [![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day145_MCmax30DMR_Tpercmeth_hist.jpg)](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day145_MCmax30DMR_Tpercmeth_hist.jpg)

- Ran ANOVA and saved summary result tables:
	- all ambient samples: [amb_MCmax30_aov_modelsumm.csv](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/MCmax30_asinT_groupStats/amb_MCmax30_aov_modelsumm.csv)
	- day 10 samples: [day10_MCmax30_aov_modelsumm.csv](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/MCmax30_asinT_groupStats/day10_MCmax30_aov_modelsumm.csv)
	- day 135 samples: [day135_MCmax30_aov_modelsumm.csv](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/MCmax30_asinT_groupStats/day135_MCmax30_aov_modelsumm.csv)
	- day 145 samples: [day145_MCmax30_aov_modelsumm.csv](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/MCmax30_asinT_groupStats/day145_MCmax30_aov_modelsumm.csv) 
 
- Plotted heatmaps and violinplots of DMRs < 0.1 ANOVA p.value:
	- all ambient samples over time heatmap:  [![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/amb_MCmax30DMR_Taov0.1_heatmap.jpg)](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/amb_MCmax30DMR_Taov0.1_heatmap.jpg)
	- all ambient samples over time violinplots:  [![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/amb_MCmax30DMR_Taov0.1_boxplots.jpg)](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/amb_MCmax30DMR_Taov0.1_boxplots.jpg)
	- day 10 samples heatmap:  [![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day10_MCmax30DMR_Taov0.1_heatmap.jpg)](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day10_MCmax30DMR_Taov0.1_heatmap.jpg)
	- day 10 samples violinplots:  [![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day10_MCmax30DMR_Taov0.1_boxplots.jpg)](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day10_MCmax30DMR_Taov0.1_boxplots.jpg)
	- day 135 samples heatmap:  [![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day135_MCmax30DMR_Taov0.1_heatmap.jpg)](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day135_MCmax30DMR_Taov0.1_heatmap.jpg)
	- day 135 samples violinplots:  [![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day135_MCmax30DMR_Taov0.1_boxplots.jpg)](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day135_MCmax30DMR_Taov0.1_boxplots.jpg)
	- day 145 samples heatmap:  [![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day145_MCmax30DMR_Taov0.1_heatmap.jpg)](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day145_MCmax30DMR_Taov0.1_heatmap.jpg)
	- day 145 samples violinplots:  [![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day145_MCmax30DMR_Taov0.1_boxplots.jpg)](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/MCmax30_asinT_groupStats/day145_MCmax30DMR_Taov0.1_boxplots.jpg)

- created DMR bedfiles:
	- used this jupyter notebook ([20191102_DMRfind_allEPI_30bp.ipynb](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20191102_DMRfind_allEPI_30bp.ipynb)) to do this 
	- all ambient samples: [amb_AllTimes\_DMR250bp\_MCmax30\_cov5x\_rms\_results\_collapsed\_AOV0.1.DMR.bed](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/MCmax30_asinT_groupStats/amb_AllTimes_DMR250bp_MCmax30_cov5x_rms_results_collapsed_AOV0.1.DMR.bed)
	- day 10 samples: [day10\_AllpH_DMR250bp\_MCmax30\_cov5x\_rms\_results\_collapsed\_AOV0.1.DMR.bed](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/MCmax30_asinT_groupStats/day10_AllpH_DMR250bp_MCmax30_cov5x_rms_results_collapsed_AOV0.1.DMR.bed)
	- day 135 samples: [day135\_AllpH_DMR250bp\_MCmax30\_cov5x\_rms\_results\_collapsed\_AOV0.1.DMR.bed](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/MCmax30_asinT_groupStats/day135_AllpH_DMR250bp_MCmax30_cov5x_rms_results_collapsed_AOV0.1.DMR.bed)
	- day 145 samples: [day145\_AllpH_DMR250bp\_MCmax30\_cov5x\_rms\_results\_collapsed\_AOV0.1.DMR.bed](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/MCmax30_asinT_groupStats/day145_AllpH_DMR250bp_MCmax30_cov5x_rms_results_collapsed_AOV0.1.DMR.bed) 

## functional analysis
### Match DMRs to genomic features
- did bedtools intersect on DMR bedfiles above and new annotation
	- gff file: [Panopea-generosa-vv0.74.a4-merged-2019-10-07-4-46-46.gff3](https://gannet.fish.washington.edu/Atumefaciens/20190928_Pgenerosa_v074.a4_gensas_annotation/Panopea-generosa-vv0.74.a4-merged-2019-10-07-4-46-46.gff3)
	- jupyter notebook: [20191102_DMR_functional_analysis.ipynb](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20191102_DMR_functional_analysis.ipynb)
	- created tab separated tables with the following columns:
		1. DMR chromosome
		2. DMR start position
		3. DMR end position
		4. number of DMS in DMR
		5. genomic feature
		6. Pgen gene 
		7. Uniprot entry ID
		8. Uniprot entry name
		9. Protein name
		10. GO IDs
		11. GO Terms	
	- [amb_AllTimes.GO.txt](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20191102_anno/amb_AllTimes.GO.txt) 
	- [day10_AllTimes.GO.txt](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20191102_anno/day10_AllpH.GO.txt) 
	- [day135_AllTimes.GO.txt](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20191102_anno/day135_AllpH.GO.txt) 
	- [day145_AllTimes.GO.txt](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20191102_anno/day145_AllpH.GO.txt) 
- DMRs from each group comparison x number of features  facetted by feature type [![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/20191102_anno/features_histograms.jpg)](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/20191102_anno/features_histograms.jpg)
- Feature type x number of features facetted by group comparison [![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/20191102_anno/features_histograms_group_facet.jpg)](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/20191102_anno/features_histograms_group_facet.jpg)
- Stacked bar plot showing the proportion of features that DMRs from each comparison fall into [![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/20191102_anno/features_stacked_bar_plot.jpg)](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/20191102_anno/features_stacked_bar_plot.jpg)


