---
layout: post
title: Apr 15 - May 3, 2019  Geoduck Broodstock hemolymph WGBS
---


# Bismark alignments 

I ran [this job](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/scripts/20190415_BismarkAlignAS1.2I60_Pgenr070_10K.sh) on mox to align reads to Pgenerosa_v071. 

- Raw fastqs live [here](http://owl.fish.washington.edu/nightingales/P_generosa/) and FastQC generated files are [here](http://gannet.fish.washington.edu/Atumefaciens/20190415_pgen_fastqc_wgbs_shelly/)

- Sam ran FastQC [here](https://robertslab.github.io/sams-notebook/2019/04/15/FastQC-WGBS-Sequencing-Data-from-Genewiz-Received-20190408.html)

- Sam [mentioned](https://github.com/RobertsLab/resources/issues/354) Genewiz recommends "Trimming 10 bases from the beginning of both R1 and R2 following adapter trimming eliminates the majority of Adaptase tails."

### Trimming

- Instead of running trimmomatic or following Genewiz's recommendations, I tried just doing a crude trim removing the first 6 characters (which seemed to be lower quality)from each read before running the alignments. [see mox job for code](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/scripts/20190415_BismarkAlignAS1.2I60_Pgenr070_10K.sh)

### running bismark aligment and methylation extractor

- I ran bismark with score_min L,0,-1.2 and insert size minimum of 60bp (-I 60 ) on trimmed and on untrimmed reads to see if the alignments are different 
- [Here is the overall Bismark alignment summary report](http://gannet.fish.washington.edu/metacarcinus/Pgenerosa/20190415_10K/bismark_summary_report.html) for trimmed and untrimmed reads

### running cytosine coverage

- because I didn't include the genome-wide cytosine coverage option when I initially ran bismark, I ran coverage2cytosine after with [this script on mox](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/scripts/20190430_BismarkCoverage2Cytosine.sh)

# Calculating coverage
I adapted [Sam's new coverage script](https://github.com/RobertsLab/code/blob/master/notebooks/sam/20190418_cvir_mbd_coverage_comparison.ipynb) to work on my data in this jupyter notebook: [20190503\_Pgnr\_comparison.ipynb](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/WGBS_conditionedBroodstockHemolymph/20190503_Pgnr_comparison.ipynb).

It worked! And generated this plot:  
![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/WGBS_conditionedBroodstockHemolymph/20190503_coverage_analysis/20190503_Pgenr_cov_comparison.png)

Coverage of genome-wide cytosines is surprising good. Tank 2 got about 66M reads and Tank 3 got about 50M reads. The genome is about 2.4gb. If the genome was evenly fragmented at 150bp and got even coverage, you would need 16M reads to cover the genome 1x. So it's possible the the depth these libraries got were able to to cover most cytosines. 

# Next steps:

### Run MethylKit
- As a first pass, run untrimmed alignments through methylkit to see what's different 

### Trim properly and rerun alignments and coverage analysis
I should probably try the alignments again with the recommended trimming and see what happens

### Check unmapped reads
I don't think that Bismark outputs unmapped reads unless you specify it in your initial code:
[https://www.bioinformatics.babraham.ac.uk/projects/bismark/](https://www.bioinformatics.babraham.ac.uk/projects/bismark/)

But when I run alignments again, I'll specify unmapped reads be output .

 