---
layout: post
title: Wed. Sept. 25, Validating Juv. low pH DMRs
category: Geoduck response to low pH
---


I realized there are differences in percent methylation between allc files, Steven's 5x destranded bed files, and original Bismark coverage files:

- see jupyter notebook: [20190925\_allc\_v\_5xbed_v\_.cov.gz.ipynb](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20190925_allc_v_5xbed_v_.cov.gz.ipynb)
- see [Mon. Sept 23 post](https://shellytrigg.github.io/182th-post/)

These differences impact how we validate DMRs. We have been using 5x.bedgraphs and .bam files to look at differences in methylation across samples and see if we believe when a DMR is called. However, if when a DMR is called and it was based on data that was further filtered down than the .bam or 5x.bedgraph files (e.g. by MAPQ score), then we may be validating with poor quality data. And we may not be seeing differences in methylation states across samples if poor quality data is included.

		
Track 1 = bedgraph from bismark (not filtered for coverage or mapq)
Track 2 = 5x destranded bedgraph from steven
Track 3/4 = deduplicated bam from bismark

Track 5 = 5x bedgraph from allc file (filtered for MAPQ 30)
Track 6 = bam file filtered for MAPQ 30 and --min-num-ch 3 and --max-mch-level 0.7

align reads to bismark
deduplicate and methylation extractor

current DMR pipeline:
generate allc files:
	- min-mapq = 30
dmr caller:
	- 5x coverage


	
Hollies pipeline:
	- .cov files from bismark are filtered for
		1. 5x coverage 
		2. covered in all samples 
