---
layout: post
title: Wed. Feb 20, 2019,  Geoduck genome versions, histology scoring, pt. whitney supplies 
---

### Geoduck gonad histology scoring:
- worked with Kaitlyn to figure out scoring for geoduck gonad histology
	- we looked at [Robert Marshall's dissertation](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&ved=2ahUKEwiMlqfk-8vgAhXYJjQIHcEcD8oQFjAAegQIAxAC&url=https%3A%2F%2Fopen.library.ubc.ca%2Fmedia%2Fdownload%2Fpdf%2F24%2F1.0072611%2F1&usg=AOvVaw1uPkvORfUA6NysyVNEGy4O)(he was in the Pearce lab at UBC), Grace's geoduck gonad project, and Molly's manuscript (from Brent) for some criteria on scoring . 
	- so far, Kaitlyn has recorded the data [here](https://docs.google.com/spreadsheets/d/1jj43XdeN0fTO_mcLKc2ribEWlCDoEJK4n0n5iTbIkco/edit?usp=sharing)	
	- for pictures and to find a scope with a ruler for measuring oocyte diameter, I contacted Lindsay in Jackie's lab to see if we can use theirs.
	
### Collected supplies to bring to Pt.Whitney tomorrow
- heath stacks
- liquid nitrogen in the dry shipper
- pipette tips
- DI water
- Mercuric chloride waste container
- microfuge tube racks (3)
- 1 box large gloves

### Comparing alignments with different P.generosa genome versions

- I aligned 4 juvenile geoduck OA RRBS PE datasets to two smaller versions of the P.generosa genome using these scripts:
	- [https://github.com/shellywanamaker/Shelly_Pgenerosa/blob/master/scripts/20190214_bmrkPgenrGenm070_10K.sh](https://github.com/shellywanamaker/Shelly_Pgenerosa/blob/master/scripts/20190214_bmrkPgenrGenm070_10K.sh)
	- [https://github.com/shellywanamaker/Shelly_Pgenerosa/blob/master/scripts/20190214_bmrkPgenrGenm070_30K.sh](https://github.com/shellywanamaker/Shelly_Pgenerosa/blob/master/scripts/20190214_bmrkPgenrGenm070_30K.sh)
	- [https://github.com/shellywanamaker/Shelly_Pgenerosa/blob/master/scripts/20190214_BismarkAlignAS1.2I60_Pgenr070_10K.sh](https://github.com/shellywanamaker/Shelly_Pgenerosa/blob/master/scripts/20190214_BismarkAlignAS1.2I60_Pgenr070_10K.sh)
	- [https://github.com/shellywanamaker/Shelly_Pgenerosa/blob/master/scripts/20190214_BismarkAlignAS1.2I60_Pgenr070_30K.sh](https://github.com/shellywanamaker/Shelly_Pgenerosa/blob/master/scripts/20190214_BismarkAlignAS1.2I60_Pgenr070_30K.sh)

- I am still copying the output data from mox to gannet, but it will be here:
	- /web/metacarcinus/Pgenerosa/20190214_10K
	- /web/metacarcinus/Pgenerosa/20190214_30K

- Using the bismark summary files from the 20190214 analyses and the file from 20181101 analysis, [I plotted the sample alignments and cytosine methylation info](https://github.com/shellywanamaker/Shelly_Pgenerosa/blob/master/analyses/CompareGenomeVersions/CompareGenomeVersions.md). 
![](https://raw.githubusercontent.com/shellywanamaker/Shelly_Pgenerosa/master/analyses/CompareGenomeVersions/CompareGenomeVersions_files/figure-markdown_github/unnamed-chunk-9-1.png)

![](https://raw.githubusercontent.com/shellywanamaker/Shelly_Pgenerosa/master/analyses/CompareGenomeVersions/CompareGenomeVersions_files/figure-markdown_github/unnamed-chunk-13-1.png)

- My hunch is that the 30K contig genome is a good middle ground. [Entire R analysis here](https://github.com/shellywanamaker/Shelly_Pgenerosa/tree/master/analyses/CompareGenomeVersions)



