---
layout: post
title: Aug 15 - Sept 9, 2019 Salmon-sea lice BS seq analysis
category: Salmon epigenomics
---


## Sea lice bismark alignments:
- mox script here: [https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20190814_BmrkCalig.sh](https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20190814_BmrkCalig.sh)
- data here: [https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20190814_Calig/](https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20190814_Calig/)
- [bismark summary report](https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20190814_Calig/bismark_summary_report.html) shows limited mehtylation as discussed on slack [here](https://caliguslife.slack.com/archives/CHD7GE7LJ/p1567532440001700). This isn't that surprising as the [global DNA methylation ELISA data](https://shellywanamaker.github.io/12th-post/) showed Female 1 only 0.56% methylated and Female 2 3.38% methylated. We had tried WGBS with high-ish coverage to overcome the limited methylation
	- Sequencing depth:
		- aimed for 240M reads/sample
		- [Female 1](https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20190814_Calig/Sealice_F1_S20_R1_001_val_1_bismark_bt2_PE_report.html): 225.7
		- [Female 2](https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20190814_Calig/Sealice_F2_S22_R1_001_val_1_bismark_bt2_PE_report.html): ~163.7M 
	- Methylated CpGs:
		- [Female 1](https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20190814_Calig/Sealice_F1_S20_R1_001_val_1_bismark_bt2_PE_report.html): ~2.15M
		- [Female 2](https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20190814_Calig/Sealice_F2_S22_R1_001_val_1_bismark_bt2_PE_report.html): ~1.33M
- **Next steps:**
	- still may be worth seeing what CpGs are present and if they are overlapping (if any have 5x coverage)
	- Maybe try ATAC-seq or different method to more sensitively detect the methylated DNA      

## Salmon bismark alignments:

#### Dealing with unplaced sequences in genome
- Aug. 14 alignments were to a genome that contained thousands of unplaced sequences:
	- figured this out from bamqc report plots
		- example:
		- [![](https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20190814_Salmo/16C_26psu_1_S13.sorted_stats/images_qualimapReport/genome_coverage_across_reference.png)](https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20190814_Salmo/16C_26psu_1_S13.sorted_stats/images_qualimapReport/genome_coverage_across_reference.png) 
	- see full preliminary bamqc reports
		- [16C_26psu_1_S13](https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20190814_Salmo/16C_26psu_1_S13.sorted_stats/qualimapReport.html)
		- [16C_26psu_1_S13](https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20190814_Salmo/16C_26psu_1_S13.sorted_stats/qualimapReport.html)
		- [16C_26psu_2_S14](https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20190814_Salmo/16C_26psu_2_S14.sorted_stats/qualimapReport.html)
		- [16C_26psu_3_S15](https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20190814_Salmo/16C_26psu_3_S15.sorted_stats/qualimapReport.html)
		- [16C_26psu_4_S16](https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20190814_Salmo/16C_26psu_4_S16.sorted_stats/qualimapReport.html)

- Talked with Steven and determined these unplaced sequences need to be removed and alignments redone
- Removing unplaced sequencings from genome:
	- jupyter notebook here: [https://github.com/shellywanamaker/Salmon_sealice/blob/master/jupyter/Remove_unplaced_genome_seqs.ipynb](https://github.com/shellywanamaker/Salmon_sealice/blob/master/jupyter/Remove_unplaced_genome_seqs.ipynb) 
- copied new genome without unplaced sequences to mox 
	- new genome fasta here: [https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/GENOMES/v2/RefSeq/GCF_000233375.1_ICSASG_v2_genomic.fa](https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/GENOMES/v2/RefSeq/GCF_000233375.1_ICSASG_v2_genomic.fa)
	- mox script to build bismark genome with new genome: [https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20190828_BmrkBld_Salmo1-29MT.sh](https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20190828_BmrkBld_Salmo1-29MT.sh) 
- mox script here: [https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20190828_BmrkBld_Salmo1-29MT.sh](https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20190828_BmrkBld_Salmo1-29MT.sh)
	- new bismark genome here: [https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/GENOMES/v2/RefSeq/chr1-29MT/](https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/GENOMES/v2/RefSeq/chr1-29MT/)

#### Alignments to bismark genome without unplaced sequences
- mox script here: [https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20190828_BmrkAlg_Salmo.sh](https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20190828_BmrkAlg_Salmo.sh)
- data here: [https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20190828_Salmo/](https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20190828_Salmo/)
- bismark summary report here: [https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20190828_Salmo/bismark_summary_report.html](https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20190828_Salmo/bismark_summary_report.html)
- multisample BamQC report here:[https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20190828_Salmo/multisampleBamQcReport.html](https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20190828_Salmo/multisampleBamQcReport.html)

#### DMR analysis
- ran preliminary DMR analysis with methylpy
	- mox script here:[https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20190907_DMRfindSalmo.sh](https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20190907_DMRfindSalmo.sh)
	- data here:[https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20190907_Salmo/](https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20190907_Salmo/)
- **Next steps:**
	- no DMRs detected with current settings so need to rerun DMRfind with more liberal parameters 