---
layout: post
title: Nov. 3-5, Salmon DMR analysis 
category: Salmon epigenomics
---

## Find DMRs 
- I updated my DMR analysis by including reads with MAPQ score of >= 20
- Using methylpy pipeline, I 
	- created allc files
		- mox script here: [20191030_DMRfindMAPQ20_Salmo.sh](https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20191030_DMRfindMAPQ20_Salmo.sh)
		- allc files here: [https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20191030/](https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20191030/)
	- created filtered bam files for QC
		- mox script here: [20191030_DMRfindMAPQ20_SalmoPt2.sh](https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20191030_DMRfindMAPQ20_SalmoPt2.sh) 
		- filtered bam files here: [https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20191030/](https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20191030/)
	- ran DMRfind to identify within sample DMRs that show methylation at a significantly different level than would be expected by genetic variation alone  
		-  mox script here: [20191030_DMRfindMAPQ20_SalmoPt2.sh](https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20191030_DMRfindMAPQ20_SalmoPt2.sh)
		-  DMR results here: [https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20191030/](https://gannet.fish.washington.edu/metacarcinus/Salmo_Calig/analyses/20191030/)

## Filter DMRs
- filtered DMRs for those showing >=5x coverage in 3/4 samples per group. I didn't use this cutoff for the control samples since these each only has two individuals.
	- R script here: [DMR_cov_in_0.75_SamplesPerCategory.Rmd](https://github.com/shellytrigg/Salmon_sealice/blob/master/analyses/DMR_cov_in_0.75_SamplesPerCategory/DMR_cov_in_0.75_SamplesPerCategory.Rmd)
	- R project here: [DMR_cov_in_0.75_SamplesPerCategory.Rproj](https://github.com/shellytrigg/Salmon_sealice/blob/master/analyses/DMR_cov_in_0.75_SamplesPerCategory/DMR_cov_in_0.75_SamplesPerCategory.Rproj)
	- filtered DMRs here: [DMR250bp_MCmax25_cov5x_rms_results_filtered.tsv](https://github.com/shellytrigg/Salmon_sealice/blob/master/analyses/DMR_cov_in_0.75_SamplesPerCategory/DMR250bp_MCmax25_cov5x_rms_results_filtered.tsv)

## Group stats on DMRs
- R script: [MCmax25_asinT_groupStats.Rmd](https://github.com/shellytrigg/Salmon_sealice/blob/master/analyses/MCmax25_asinT_groupStats/MCmax25_asinT_groupStats.Rmd)
- R project: [MCmax25_asinT_groupStats.Rproj](https://github.com/shellytrigg/Salmon_sealice/blob/master/analyses/MCmax25_asinT_groupStats/MCmax25_asinT_groupStats.Rproj)

### Data preprocessing:
- To determine if there is a significant experimental effect on DMRs I first arcsin(sqrt) transformed the [percent methylation data](https://github.com/shellytrigg/Salmon_sealice/blob/master/analyses/DMR_cov_in_0.75_SamplesPerCategory/DMR250bp_MCmax25_cov5x_rms_results_filtered.tsv)
- **BEFORE** transformation percent methylation distribution: [![](https://raw.githubusercontent.com/shellytrigg/Salmon_sealice/master/analyses/MCmax25_asinT_groupStats/DMR_percmeth_hist.jpg)](https://raw.githubusercontent.com/shellytrigg/Salmon_sealice/master/analyses/MCmax25_asinT_groupStats/DMR_percmeth_hist.jpg)
- - **AFTER** transformation percent methylation distribution: [![](https://raw.githubusercontent.com/shellytrigg/Salmon_sealice/master/analyses/MCmax25_asinT_groupStats/DMR_Tpercmeth_hist.jpg)](https://raw.githubusercontent.com/shellytrigg/Salmon_sealice/master/analyses/MCmax25_asinT_groupStats/DMR_Tpercmeth_hist.jpg)

### Run ANOVA
1. I ran a 1 way ANOVA to test if sea lice infestation had an effect
	- 1 way ANOVA results summary table here: [DMR_MCmax25_1wayAOV_infest_modelsumm.csv](https://github.com/shellytrigg/Salmon_sealice/blob/master/analyses/MCmax25_asinT_groupStats/DMR_MCmax25_1wayAOV_infest_modelsumm.csv) 
	- 3 DMRs showed a significant infestation effect at 1way ANOVA p-value of < 0.01 [![](https://raw.githubusercontent.com/shellytrigg/Salmon_sealice/master/analyses/MCmax25_asinT_groupStats/DMR_MCmax25DMR_Taov0.01InfestPercMeth.jpg)](https://raw.githubusercontent.com/shellytrigg/Salmon_sealice/master/analyses/MCmax25_asinT_groupStats/DMR_MCmax25DMR_Taov0.01InfestPercMeth.jpg)
	- 13 DMRs showed a significant infestation effect at 1way ANOVA p-value of < 0.05 [![](https://raw.githubusercontent.com/shellytrigg/Salmon_sealice/master/analyses/MCmax25_asinT_groupStats/DMR_MCmax25DMR_Taov0.05InfestPercMeth.jpg)](https://raw.githubusercontent.com/shellytrigg/Salmon_sealice/master/analyses/MCmax25_asinT_groupStats/DMR_MCmax25DMR_Taov0.05InfestPercMeth.jpg)
	- 24 DMRs showed a significant infestation effect at 1way ANOVA p-value of < 0.1 
[![](https://raw.githubusercontent.com/shellytrigg/Salmon_sealice/master/analyses/MCmax25_asinT_groupStats/DMR_MCmax25DMR_Taov0.1InfestPercMeth.jpg)](https://raw.githubusercontent.com/shellytrigg/Salmon_sealice/master/analyses/MCmax25_asinT_groupStats/DMR_MCmax25DMR_Taov0.1InfestPercMeth.jpg)  
Heatmap of 24 DMRs that showed a significant infestation effect at 1way ANOVA p-value of < 0.1. Heatmap key:Column color bar in heat maps below: light pink = 16C_26psu, dark pink = 16C_32psu, light green = 8C_26psu, dark green = 8C_32psu, magenta = CTRL_16C_26psu, cyan = CTRL_8C_26psu. heatmap color: Red = more methylation, blue = no methylation, black = no data.
[![](https://raw.githubusercontent.com/shellytrigg/Salmon_sealice/master/analyses/MCmax25_asinT_groupStats/DMR_MCmax25DMR_Taov0.1_infest_heatmap.jpg)](https://raw.githubusercontent.com/shellytrigg/Salmon_sealice/master/analyses/MCmax25_asinT_groupStats/DMR_MCmax25DMR_Taov0.1_infest_heatmap.jpg)

2. For the 24 DMRs that showed a significant infestation effect at 1way ANOVA p-value of < 0.1, I ran a two way ANOVA on their percent methylation to see if temperature, salinity, or their interaction showed a significant effect.
	- 2 way ANOVA results summary table here: [DMR_MCmax25_2wayAOV_TxS_modelsumm.csv](https://github.com/shellytrigg/Salmon_sealice/blob/master/analyses/MCmax25_asinT_groupStats/DMR_MCmax25_2wayAOV_TxS_modelsumm.csv)
	- Tukey HSD results summary table here: [DMR_MCmax25_DMR_TukHSD_TxS_modelsumm.csv](https://github.com/shellytrigg/Salmon_sealice/blob/master/analyses/MCmax25_asinT_groupStats/DMR_MCmax25_DMR_TukHSD_TxS_modelsumm.csv)
	- 3 DMRs showed a significant Salinity effect (at adj.p.value < 0.1)
[![](https://raw.githubusercontent.com/shellytrigg/Salmon_sealice/master/analyses/MCmax25_asinT_groupStats/DMR_MCmax25DMR_Taov0.1SalPercMeth.jpg)](https://raw.githubusercontent.com/shellytrigg/Salmon_sealice/master/analyses/MCmax25_asinT_groupStats/DMR_MCmax25DMR_Taov0.1SalPercMeth.jpg)
	- 3 DMRs showed a significant temperature effect (at adj.p.value < 0.1)
[![](https://raw.githubusercontent.com/shellytrigg/Salmon_sealice/master/analyses/MCmax25_asinT_groupStats/DMR_MCmax25DMR_Taov0.1TempPercMeth.jpg)](https://raw.githubusercontent.com/shellytrigg/Salmon_sealice/master/analyses/MCmax25_asinT_groupStats/DMR_MCmax25DMR_Taov0.1TempPercMeth.jpg)
	- 6 DMRs showed a significant temperature effect (at adj.p.value < 0.1)
[![](https://raw.githubusercontent.com/shellytrigg/Salmon_sealice/master/analyses/MCmax25_asinT_groupStats/DMR_MCmax25DMR_Taov0.1TxSPercMeth.jpg)](https://raw.githubusercontent.com/shellytrigg/Salmon_sealice/master/analyses/MCmax25_asinT_groupStats/DMR_MCmax25DMR_Taov0.1TxSPercMeth.jpg)

## Next steps
- validate 24 DMRs that showed a significant infestation effect at 1way ANOVA p-value of < 0.1 in IGV
	- generate ANOVA filtered bed file 
	- compare to filtered bam files linked above 
- find out where in the genome these DMRs are