---
layout: post
title: Nov. 7-9, Geoduck DMR feature analysis 
---

## Generate appropriate background
- I previously used within-sample DMRs filtered for coverage in 3/4 individuals/group as the background. 
	- HOWEVER, these did not include all sites that had the potential to be methylated 
	- To create a more inclusive background, I need to look at all CG sites considered prior to determining within-sample DMRs

- Within-sample DMRs were determined by: 
	- having at least 3 differentially methylated sites (DMS)
		- DMS were determined by:
			- having 5x coverage
				- if sites are within a 30bp window their counts can be combined
			- passing an RMS test significance threshold of 0.01  
	- within a max distance of 250bp

- **ATTEMPT 1:**  If I remove the significance threshold (and set --sig-cutoff to 1 instead of 0.01) for DMS, than I should get results from all sites considered for DMS. 
	- I attempted this using this mox script:  [20191108_DMRfindAllEPInoSig.sh](https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20191108_DMRfindAllEPInoSig.sh)
		- RESULTS: [Resulting files](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20191108/) indicate some other filtering is happening because because the number of regions is the nearly the same as when I ran DMRfind with significance level set to 0.01. I had expected a much longer list of regions.

- **ATTEMPT 2:**  To address if the software only consider CG sites with a value in the methylation counts column of the allc file I created new allc files with the coverage counts column also as the methylation counts column (assuming 100% methylation at every site). 
	- I attempted this using this mox script:  [20191108_DMRfindAllEPItotCounts.sh](https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20191108_DMRfindAllEPItotCounts.sh)
		-  allc files with the coverage counts column also as the methylation counts here: [https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20191108/tot_counts_as_mC_counts/](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20191108/tot_counts_as_mC_counts/)
		- RESULTS: [Resulting files](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20191108/tot_counts_as_mC_counts/) indicate some other filtering is happening because because the number of regions is the nearly the same as when I ran DMRfind with significance level set to 0.01. I had expected a much longer list of regions.

- **ATTEMPT 3:**  To adjust other parameters in attempt to remove all filtering, I tried the following settings in this mox script here:  [20191108_DMRfindAllEPItotCountsRes1.sh](https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20191108_DMRfindAllEPItotCountsRes1.sh) on ambient samples only as a test:
	- --resid-cutoff 1
	- --min-tests 1
	- --num-sims 1
	- --sig-cutoff 1 	
	- RESULTS: [resulting files](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20191108/tot_counts_as_mC_counts/set_resd_1/) indicate certain criteria in the software are not met as output contains only a header

### 4th times the charm!
- **ATTEMPT 4:** Don't use DMRfind, just filter allc files for 5x coverage and find all CG sites in at least 3/4 samples
	1. Filter allc files for sites with 5x coverages and CG context
		- mox script here: [20191108_filterAllc5x.sh](https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20191108_filterAllc5x.sh)
		- input files here: [https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20190925/](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20190925/)
		- output files (and slurm file) here: [https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20191108/5x_allc/](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20191108/5x_allc/) 
	2. Filter CpG sites with coverage in 3/4 individuals per experimental group, concatenate group sites for each comparison (all ambient samples, all day 10 samples, all day 135 samples, and all day 145 samples), and create bed files to intersect with GFF
		- mox script here: [20191108_combine_filt_allc.sh](https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20191108_combine_filt_allc.sh)
			- I successfully implemented a bash loop within a loop to call different sets of files to filter sites for each experimental group! Found [this post really helpful](https://stackoverflow.com/questions/35543088/iterating-over-list-of-arrays-in-bash) 
			- Also added 'set -ex' to my bash script so that the slurm file records every command as it is called. [slurm file here](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20191108/combine_filt_allc/slurm-1460011.out)
		- Input files here: [https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20191108/5x_allc/](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20191108/5x_allc/) 
			- I unzipped the input files before running the script on mox because I kept getting zcat related errors (see slurm scripts: [slurm-1459988.out](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20191108/combine_filt_allc/slurm-1459988.out) and [slurm-1459990.out](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20191108/combine_filt_allc/slurm-1459990.out)). 
		- Output files here: [https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20191108/combine_filt_allc/](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20191108/combine_filt_allc/)
	3. Intersect bedfiles with GFF to get genomic features for sites from each comparison 
		- mox script here: [20191109_EPIbkgdFeatures.sh](https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20191109_EPIbkgdFeatures.sh)
		- GFF file here: [Panopea-generosa-vv0.74.a4-merged-2019-10-07-4-46-46.gff3](https://gannet.fish.washington.edu/Atumefaciens/20190928_Pgenerosa_v074.a4_gensas_annotation/Panopea-generosa-vv0.74.a4-merged-2019-10-07-4-46-46.gff3)
		- Input files here:
			- [all_amb_bkgd.bed](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20191108/combine_filt_allc/all_amb_bkgd.bed)
			- [all_day10_bkgd.bed](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20191108/combine_filt_allc/all_day10_bkgd.bed) 
			- [all_day135_bkgd.bed](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20191108/combine_filt_allc/all_day135_bkgd.bed) 
			- [all_day145_bkgd.bed](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20191108/combine_filt_allc/all_day145_bkgd.bed)  
		- Output files here: [https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20191109/](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20191109/) 
  
### Plot background vs. sig. DMR features 
I updated [the Rmarkdown file](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20191102_anno/20191102_anno.Rmd) I used [previously](https://shellytrigg.github.io/211th-post/) to generate a bar plot showing the proportion of features that significant DMRs vs. background sites fall into.

[![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/20191102_anno/BkgdvSigDMRsxfeatures_PropBarplot_GroupFacet.jpg)](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/20191102_anno/BkgdvSigDMRsxfeatures_PropBarplot_GroupFacet.jpg)

**RESULTS:**  There are no strong differences between significant DMRs and background CpG sites in the proportion of features they overlap with, _except:_ 
	
- Day145 comparison where CDS and exon features are under-represented, and repeat_regions are over-represented.
  
- CDS and exon are slightly over-represented in Day135 and all ambient sample comparisons

## Next steps
- Look deeper into repeat regions (there are ~6 different catagories, so can see if any are particularly different)
- For DMRs no in features, check the nearest features
- Compare genes with DMRs to the ones identified by Hollie's method
- Continue GO analysis
	- generate appropriate GO background to use for each comparison
	- run TopGO
- Go through manuscript methods section 
	- update new stuff
	- add comments to areas of concern