---
layout: post
title: Tues. Oct. 27, Salmon RNAseq
tags: salmon RNAseq HISAT2 nf-core nextflow
catagory: Salmon 
---

I'm processing Salmon RNAseq data that match my Salmon methyltion data so I can perform an integrated analysis of these. 

I'm using the nf-core RNAseq pipeline. 

## On Mox:
I started installing it on Mox and got to the initial pipeline test, but this failed because it couldn't allocated enough memory (which I suspect is because I was running it from the command prompt rather than as a script). Unfortunately the nodes were all being used, so I decided to try running it on Roadrunner.

## On Roadrunner: 
This was generally a smooth installation process. I had to install docker because I couldn't get it to work with conda. I had one hangup with docker and it was a permissions thing that I was able to resolve following a github issue (noted in log file from installation). The RNAseq pipeline test completed without errors and the RNAseq pipeline is currently running on the raw data on Roadrunner

#### Log file from installation here:
[install_nf-core_Nextflow.log](https://github.com/shellytrigg/Salmon_sealice/blob/master/analyses/20201027/install_nf-core_Nextflow.log)

### Running RNAseq pipeline on Salmon data

- had to adjust the max memory allocated to processes because this computer only has 48GB available
	- I did this by setting the --max-memory parameter to '48.GB'

```
srlab@roadrunner:~/Documents/Shelly/Ssalar nextflow run nf-core/rnaseq -profile docker \
--max_memory '45.GB'
--reads '*_{1,2}.fastq.gz' \
--aligner hisat2 \
--fasta GCF_000233375.1_ICSASG_v2/GCF_000233375.1_ICSASG_v2_genomic.fa \
--gtf GCF_000233375.1_ICSASG_v2/GCF_000233375.1_ICSASG_v2_genomic.gtf \
--gff GCF_000233375.1_ICSASG_v2/GCF_000233375.1_ICSASG_v2_genomic.gff \
--bed12 GCF_000233375.1_ICSASG_v2/GCF_000233375.1_ICSASG_v2_MASTER.bed \
--saveReference \
--saveTrimmed \
--saveUnaligned \
--saveAlignedIntermediates 
```


~12 hours to complete using 45gb ram


