revisiting salmon alignments

- trimming data based on bismark recommendations
- testing new alignment parameters
	- previously tested different alignment score thresholds
	- now testing 
		- single end mode vs. PE mode
		- global vs local alignment mode 
			- more info [https://github.com/FelixKrueger/Bismark/issues/265](https://github.com/FelixKrueger/Bismark/issues/265) 
			- explore what causes the differences, and whether or not you trust the local alignments. If, for instance, all extra alignments in local mode originate from one certain repetitive location of the genome, there is probably not much added value in using it. If you find a technical justification why you would want to use local mode in your case, and this adds additional reads everywhere equally, it might be a good idea.
			- bedtools intersect -v with .deduplicated.bedGraph.gz (local) and .deduplicated.bedGraph.gz (global)
			- what do you gain, what do you lose?
			- bin genome up into 2kb bins and bedtools intersect ?
			- **compare sorted bam files and unique loci across different comparisons**

- evaluate by 
	- check reports for % alignment (unique vs. ambig)
	- %CpG methylation
	- % CHH methylation?
	- IGV of bam files

	
	
PE vs. SE
- TG doesn't matter (nearly identical numbers, largest difference is 40K reads more in SE vs. PE)
- Alignments
	- ~ 5% increase in unique alignments with SE (~700K reads)
	- ~ 3% increase in ambiguous alignments with SE
	- ~ 7% decrease in no alignments
	- percent methylation does not change
	- ~ 7% increase in duplicated reads
	- ~ 2x increase in total number of reads because SE

PE vs. SEL
- Alignments
	- ~ 20% increase in unqiue alignments (~ 3M reads)
	- ~ 7% increase in ambiguous alignments
	- ~ 28% decrease in no alignments
	- ~ 6% increase in duplicated reads
	- ~ 2.6x increase in total number of reads because SE

