---
layout: post
title: Fri. Mar 27, 2020 Geoduck new trim DMR analysis
tags: new, trim, DMR, stats, ANOVA
category: Geoduck epigenomics 
---

## Run DMRfind on new alignments
This is the follow up analysis from these posts:

- [https://shellytrigg.github.io/232th-post/](https://shellytrigg.github.io/232th-post/)
- [https://shellytrigg.github.io/235th-post/](https://shellytrigg.github.io/235th-post/)


- Copy 5x.tab files to Ostrich, Convert 5x.tab files to allc format, and Run DMRfind:  
	- script here: [https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/20200327_DMRfind.sh](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/20200327_DMRfind.sh) 
	- screen output here: [https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/screenlog.0](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/screenlog.0)
	- all files, scripts, and logs were copied to Gannet here:[https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/) 
		- 5x.tab files: [https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/merged_tab/](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/merged_tab/)	
		- allc files: [https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/)
		- DMRfind output: 
			- All ambient samples:
				- [amb_AllTimes_DMR250bp_cov5x_rms_results.tsv.gz](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/amb_AllTimes_DMR250bp_cov5x_rms_results.tsv.gz)
				- [amb_AllTimes_DMR250bp_cov5x_rms_results_collapsed.tsv](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/amb_AllTimes_DMR250bp_cov5x_rms_results_collapsed.tsv)
				- [amb_AllTimes_DMR250bp_cov5x_rms_results_collapsed.tsv.DMR.bed](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/amb_AllTimes_DMR250bp_cov5x_rms_results_collapsed.tsv.DMR.bed)
				- [amb_AllTimes_DMR250bp_cov5x_rms_results_collapsed.tsv.DMS.bed](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/amb_AllTimes_DMR250bp_cov5x_rms_results_collapsed.tsv.DMS.bed)
			- Day 10 all pH samples:
				- [day10_AllpH_DMR250bp_cov5x_rms_results.tsv.gz](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/day10_AllpH_DMR250bp_cov5x_rms_results.tsv.gz)
				- [day10_AllpH_DMR250bp_cov5x_rms_results_collapsed.tsv](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/day10_AllpH_DMR250bp_cov5x_rms_results_collapsed.tsv)
				- [day10_AllpH_DMR250bp_cov5x_rms_results_collapsed.tsv.DMR.bed](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/day10_AllpH_DMR250bp_cov5x_rms_results_collapsed.tsv.DMR.bed)
				- [day10_AllpH_DMR250bp_cov5x_rms_results_collapsed.tsv.DMS.bed](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/day10_AllpH_DMR250bp_cov5x_rms_results_collapsed.tsv.DMS.bed)
			- Day 135 all pH samples:
				- [day135_AllpH_DMR250bp_cov5x_rms_results.tsv.gz](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/day135_AllpH_DMR250bp_cov5x_rms_results.tsv.gz)
				- [day135_AllpH_DMR250bp_cov5x_rms_results_collapsed.tsv](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/day135_AllpH_DMR250bp_cov5x_rms_results_collapsed.tsv)
				- [day135_AllpH_DMR250bp_cov5x_rms_results_collapsed.tsv.DMR.bed](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/day135_AllpH_DMR250bp_cov5x_rms_results_collapsed.tsv.DMR.bed)
				- [day135_AllpH_DMR250bp_cov5x_rms_results_collapsed.tsv.DMS.bed](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/day135_AllpH_DMR250bp_cov5x_rms_results_collapsed.tsv.DMS.bed)
			- Day 145 all pH samples:
				- [day145_AllpH_DMR250bp_cov5x_rms_results.tsv.gz](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/day145_AllpH_DMR250bp_cov5x_rms_results.tsv.gz)
				- [day145_AllpH_DMR250bp_cov5x_rms_results_collapsed.tsv](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/day145_AllpH_DMR250bp_cov5x_rms_results_collapsed.tsv)
				- [day145_AllpH_DMR250bp_cov5x_rms_results_collapsed.tsv.DMR.bed](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/day145_AllpH_DMR250bp_cov5x_rms_results_collapsed.tsv.DMR.bed)
				- [day145_AllpH_DMR250bp_cov5x_rms_results_collapsed.tsv.DMS.bed](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200327/day145_AllpH_DMR250bp_cov5x_rms_results_collapsed.tsv.DMS.bed)

## Filter DMRs, Run ANOVA on DMRs, Update heatmaps

- R project here: [20200320_anno.Rproj](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20200320_anno/20200320_anno.Rproj)
- R markdown here: [DMR_stats_and_anno.Rmd](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20200320_anno/DMR_stats_and_anno.Rmd)
- DMRs filtered for coverage in 3/4 samples per treatment group:
	-[amb_AllTimes_DMR250bp_cov5x_rms_results_filtered.tsv](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20200320_anno/amb_AllTimes_DMR250bp_cov5x_rms_results_filtered.tsv)
	- [day10_AllpH_DMR250bp_cov5x_rms_results_filtered.tsv](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20200320_anno/day10_AllpH_DMR250bp_cov5x_rms_results_filtered.tsv)
	- [day135_AllpH_DMR250bp_cov5x_rms_results_filtered.tsv](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20200320_anno/day135_AllpH_DMR250bp_cov5x_rms_results_filtered.tsv)
	- [day145_AllpH_DMR250bp_cov5x_rms_results_filtered.tsv](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20200320_anno/day145_AllpH_DMR250bp_cov5x_rms_results_filtered.tsv) 
- DMRs significant at ANOVA p.val 0.05:
	- [aov\_0.05pH\_d10DMR.bed](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20200320_anno/aov_0.05pH_d10DMR.bed) 
	- [aov\_0.05pH\_d135DMR.bed](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20200320_anno/aov_0.05pH_d135DMR.bed)
	- [aov\_0.05pH\_d145DMR.bed](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20200320_anno/aov_0.05pH_d145DMR.bed)  
	- [aov\_0.05pH\_ambDMR.bed](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20200320_anno/aov_0.05pH_ambDMR.bed)
- Heatmaps of significant DMRs
	- **Ambient samples over time:**
	![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/20200320_anno/amb_MCmax25DMR_Taov0.05_GROUPmean_heatmap2.jpg)
	- **Day 10 samples**
	![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/20200320_anno/d10_MCmax25DMR_Taov0.05_GROUPmean_heatmap2.jpg)
	- **Day 135 samples**
	![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/20200320_anno/d135_MCmax25DMR_Taov0.05_GROUPmean_heatmap2.jpg)
	- **Day 145 samples**
	![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/20200320_anno/d145_MCmax25DMR_Taov0.05_GROUPmean_heatmap2.jpg)  
	- Fig for paper
	[![](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/20200320_anno/allComparisons_heatmaps.jpg)](https://raw.githubusercontent.com/shellytrigg/Shelly_Pgenerosa/master/analyses/20200320_anno/allComparisons_heatmaps.jpg)
- Other plots like data distribution before and after arcsin sqrt transformation and percent methylation plots facetted by individual DMRs can be found here: [https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20200320_anno/](https://github.com/shellytrigg/Shelly_Pgenerosa/blob/master/analyses/20200320_anno/)

## Next steps
- perform genomic feature analysis on DMRs vs background regions 


