---
layout: post
title: Fri. Mar 19-27, 2020 Geoduck EPI Trimming Updated!
tags: TrimGalore, trim, geoduck, genome, methods, bismark
category: Geoduck epigenomics 
---

## BSseq method clarification
- determined the following during conference call:
	- samples were digested with MspI but not size selected prior to library prep
	- digested genomic DNA was randomly primed during first round of PCR in library kit (see [kit schematic](https://raw.githubusercontent.com/shellywanamaker/shellywanamaker.github.io/master/images/Adaptor-tagged%20TruSeq%20DNA%20Methylation%20LIbrary%20Kit%20Workflow.png))
		- **all read start sites should be now be random and no longer at the MspI cut site**
		- we DO want to deduplicate because we do not expect reads to stack up since they are randomly primed 
	- we should follow [bismark user guide recommendations](https://htmlpreview.github.io/?https://github.com/FelixKrueger/Bismark/blob/master/Docs/Bismark_User_Guide.html#ix-notes-about-different-library-types-and-commercial-kits) for TruSeq DNA-Methylatin Kit for trimming
		- trim an extra 8bp off both ends of reads 

## New Trimming Pilot

- Script here: [20200319_TG_EPI-Test.sh](https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20200319_TG_EPI-Test.sh)
- Slurm file here: [slurm-2358892.out](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200319/slurm-2358892.out)
- Parameter description:
	- 8 bp were clipped off the beginning and end of each read using the following TG parameters:

		```
		--clip_R1 8  
		--clip_R2 8 
		--three_prime_clip_R1 8 
		--three_prime_clip_R2 8 
		```

	- adapters were either specified using the illumina sequences provided on page 45 of the Illumina adapter sequences document or were identified using the default settings
	
		```
		#specifying adapters
		--adapter AGATCGGAAGAGCACACGTCTGAAC 
		--adapter2 AGATCGGAAGAGCGTCGTGTAGGGA 
		
		#default settings did not include the --adapter and --adapter2 options
		```
- newly trimmed reads are on Gannet here [https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200319/](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200319/) and mox here:
/gscratch/scrubbed/strigg/analyses/20200319
	- reads with specified adapters trimmed:
		- [specify_a/EPI-167_S10_L002_R1_001_val_1.fq.gz](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200319/specify_a/EPI-167_S10_L002_R1_001_val_1.fq.gz) 
		- [specify_a/EPI-167_S10_L002_R2_001_val_2.fq.gz](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200319/specify_a/EPI-167_S10_L002_R2_001_val_2.fq.gz)
	- reads with TrimGalore! defaults adapter trimmed:
		- [EPI-167_S10_L002_R1_001_val_1.fq.gz](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200319/EPI-167_S10_L002_R1_001_val_1.fq.gz) 
		- [EPI-167_S10_L002_R2_001_val_2.fq.gz](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200319/EPI-167_S10_L002_R2_001_val_2.fq.gz)
- FastQC files:
	- BEFORE (raw data):
		- [EPI-167 R1](http://owl.fish.washington.edu/Athaliana/20180516_geoduck_EPI_fastqc/EPI-167_S10_L002_R1_001_fastqc.html)
		- [EPI-167 R2](http://owl.fish.washington.edu/Athaliana/20180516_geoduck_EPI_fastqc/EPI-167_S10_L002_R2_001_fastqc.html)

	- AFTER trimming with adapter specified):
		- [EPI-167 R1](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200319/specify_a/EPI-167_S10_L002_R1_001_val_1_fastqc.html)
		- [EPI-167 R2](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200319/specify_a/EPI-167_S10_L002_R2_001_val_2_fastqc.html)

	- AFTER trimming with default adapters:
		- [EPI-167 R1](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200319/EPI-167_S10_L002_R1_001_val_1_fastqc.html)
		- [EPI-167 R2](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200319/EPI-167_S10_L002_R2_001_val_2_fastqc.html)

- Aligned reads
	-  Steven's IGV session [https://raw.githubusercontent.com/sr320/nb-2020/master/P_generosa/igv/032020-igv_session.xml](https://raw.githubusercontent.com/sr320/nb-2020/master/P_generosa/igv/032020-igv_session.xml)
	- [IGV screen capture video](https://d.pr/v/cxr5GH)
	- [Steven's nb post](https://sr320.github.io/fds/)
	
- Assciated github issues:
		- [https://github.com/hputnam/Geoduck_Meth/issues/3](https://github.com/hputnam/Geoduck_Meth/issues/3)
		- [https://github.com/RobertsLab/resources/issues/860](https://github.com/RobertsLab/resources/issues/860)

## New Trimming on all data
- Pilot trimming looked good and there wasn't a big difference between specifying the exact adapter sequence or using default adapter settings so we decided to go with default
- copied all raw data to mox: [https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200320/readme.txt](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200320/readme.txt)
- Ran this script on mox: [20200320_TrimGpgnrMeth1.sh](https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20200320_TrimGpgnrMeth1.sh) 
	- [slurm file](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200320/slurm-2386816.out) 
	- Trimmed reads are here: [https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200320/TG_FASTQS/](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200320/TG_FASTQS/)
	- FastQC files are here: [https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200320/TG_FASTQS/FastQC/](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200320/TG_FASTQS/FastQC/)
	- MultiQC report is here: [https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200320/multiqc_report.html](https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200320/multiqc_report.html) 
	- OFS was updated with newly trimmed reads [https://github.com/hputnam/Geoduck_Meth/issues/4](https://github.com/hputnam/Geoduck_Meth/issues/4)

## Alignments on newly trimmed data
- Steven ran this Bismark script on mox: [https://github.com/hputnam/Geoduck_Meth/blob/master/code/03-bismark.sh](https://github.com/hputnam/Geoduck_Meth/blob/master/code/03-bismark.sh)
	- [slurm file](https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/032120-fds/slurm-2417730.out) 
	- Bismark alignments are here: [https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/032120-fds/](https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/032120-fds/)
	- Bismark summary report: [https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/032120-fds/bismark_summary_report.html](https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/032120-fds/bismark_summary_report.html)
- OFS was updated with new alignment files [https://github.com/hputnam/Geoduck_Meth/issues/4](https://github.com/hputnam/Geoduck_Meth/issues/4)


---