---
layout: post
title: Running RNAseq pipeline with reference transcript for C.Gigas Dataset #1
tags: RNAseq nextflow Klone referencetranscript Cgigas
---

## RNASeq workflow for C.Gigas Dataset #1

This workflow is fully documented in a separate post [here](https://resilience-biomarkers-for-aquaculture.github.io/ES-RNAseq-pipeline-construction-post/).

Datasets:    
- [1: Arredondo-Espinoza et al 2023 C. gigas thermotolerance](#dataset-1-arredondo-espinoza-et-al-2023)    

Data is downloaded via nextflow pipeline [fetchNGS](https://nf-co.re/fetchngs/1.12.0/) and see Shelly's example download post [here](https://resilience-biomarkers-for-aquaculture.github.io/a-fetchNGSKlone/). 

Creating a mamba environment on my user for following workflows:

```
wget https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-Linux-x86_64.sh

bash Mambaforge-Linux-x86_64.sh -b -p $HOME/mambaforge

source ~/mambaforge/bin/activate
mamba init 
## close and reopen shell for changes to take place 

mamba create -n nextflow -c bioconda -c conda-forge python=3.8 nf-core nextflow -y -q
```

Downloading [C.gigas genome](https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/963/853/765/GCF_963853765.1_xbMagGiga1.1/). Using xbMagGiga1.1 published in December 2023. Ref ID: GCF_963853765.1

```
cd /mmfs1/gscratch/scrubbed/elstrand/genomes
mkdir C_gigas

wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/963/853/765/GCF_963853765.1_xbMagGiga1.1/GCF_963853765.1_xbMagGiga1.1_genomic.fna.gz
wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/963/853/765/GCF_963853765.1_xbMagGiga1.1/GCF_963853765.1_xbMagGiga1.1_genomic.gff.gz
wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/963/853/765/GCF_963853765.1_xbMagGiga1.1/GCF_963853765.1_xbMagGiga1.1_genomic.gtf.gz
```

## Dataset 1: Arredondo-Espinoza et al 2023

Paper found [here](https://www.sciencedirect.com/science/article/pii/S1744117X23000345?via%3Dihub). Shelly downloaded this data on [20240925 with fetchNGS](https://resilience-biomarkers-for-aquaculture.github.io/a-fetchNGSKlone/). 

This study focused on the Pacific oyster, *Crassostrea gigas*, and completed a laboratory-simulated daily oscillatory thermal challenge (26 to 34 Â°C) for 30 days. 

Path to fastq files: `/mmfs1/gscratch/scrubbed/<ShellynetID>/analyses/20240925/fastq`   
Path to samplesheet created by fetchNGS: `/mmfs1/gscratch/scrubbed/<ShellynetID>/analyses/20240925/samplesheet/samplesheet.csv` 

Path to Emma's analysis: `/mmfs1/gscratch/scrubbed/<EmmanetID>/Cgigas_ArredondoEspinoza2023/`
Path to C.gigas genome: `/mmfs1/gscratch/scrubbed/elstrand/genomes/GCF_963853765.1_xbMagGiga1.1_genomic.fna.gz`

Testing this pipeline: 

```
# create screen session
screen -S rnaseq_cgigas_AE2023

# request a compute node (mem and time requests can be modified)
salloc -A srlab -p cpu-g2-mem2x -N 1 -c 1 --mem=40GB --time=12:00:00

# activate mamba environment (created at beginning of this post)
mamba activate nextflow

# update rnaseq workflow in nextflow within mamba environment
nextflow pull nf-core/rnaseq

## attach back to screen 
screen attach-session -t rnaseq_cgigas_AE2023
```

Nextflow RNAsew workflow:

```
samplesheet="/mmfs1/gscratch/scrubbed/strigg/analyses/20240925/samplesheet/samplesheet.csv"
output="/mmfs1/gscratch/scrubbed/elstrand/Cgigas_ArredondoEspinoza2023/"
genome="/mmfs1/gscratch/scrubbed/elstrand/genomes/C_gigas/GCF_963853765.1_xbMagGiga1.1_genomic.fna.gz"
gff="/mmfs1/gscratch/scrubbed/elstrand/genomes/C_gigas/GCF_963853765.1_xbMagGiga1.1_genomic.gff.gz"
gtf="/mmfs1/gscratch/scrubbed/elstrand/genomes/C_gigas/GCF_963853765.1_xbMagGiga1.1_genomic.gtf.gz"

nextflow run nf-core/rnaseq \
    -resume \
    -profile singularity \
    --input ${samplesheet} \
    --outdir ${output} \
    --gtf ${gtf} \
    --gff ${gff} \
    --fasta ${genome} \
    --trimmer fastp \
    --aligner star_salmon \
    --pseudo_aligner salmon \
    --multiqc_title rnaseq_projectitle_refgenome \
    --deseq2_vst
```

#### 2024-10-09

Trying the above without any extra flags to see if this runs properly. Will need to adjust trimming based on multiqc output.  

Error below. The samplesheet doesn't have a strandedness column.. This samplesheet.csv was generated by fetchNGS. Is there way to default to auto?

```
ERROR ~ ERROR: Validation of 'input' file failed!

 -- Check '.nextflow.log' file for details
The following errors have been detected:

* -- Entry 1: Missing required value: strandedness
* -- Entry 2: Missing required value: strandedness
* -- Entry 3: Missing required value: strandedness
* -- Entry 4: Missing required value: strandedness
```